<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
<title>Mike's NAO notebook: WWTP basic detection theory</title>

<meta property="description" itemprop="description" content="Notes exploring the factors that determine our ability to perform _basic detection_ in a WWTP setting. &#10;By basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth."/>


<!--  https://schema.org/Article -->
<meta property="article:published" itemprop="datePublished" content="2023-06-03"/>
<meta property="article:created" itemprop="dateCreated" content="2023-06-03"/>
<meta name="article:author" content="Michael R. McLaren"/>

<!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
<meta property="og:title" content="Mike&#39;s NAO notebook: WWTP basic detection theory"/>
<meta property="og:type" content="article"/>
<meta property="og:description" content="Notes exploring the factors that determine our ability to perform _basic detection_ in a WWTP setting. &#10;By basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth."/>
<meta property="og:locale" content="en_US"/>
<meta property="og:site_name" content="Mike&#39;s NAO notebook"/>

<!--  https://dev.twitter.com/cards/types/summary -->
<meta property="twitter:card" content="summary"/>
<meta property="twitter:title" content="Mike&#39;s NAO notebook: WWTP basic detection theory"/>
<meta property="twitter:description" content="Notes exploring the factors that determine our ability to perform _basic detection_ in a WWTP setting. &#10;By basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth."/>

<!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Mathematical Tools for Understanding Infectious Disease Dynamics;citation_publisher=Princeton University Press;citation_doi=10.23943/princeton/9780691155395.001.0001;citation_author=Odo Diekmann;citation_author=Hans Heesterbeek;citation_author=Tom Britton"/>
  <meta name="citation_reference" content="citation_title=The approximately universal shapes of epidemic curves in the Susceptible-Exposed-Infectious-Recovered (SEIR) model;citation_publisher=Nature Publishing Group;citation_volume=10;citation_doi=10.1038/s41598-020-76563-8;citation_author=Kevin Heng;citation_author=Christian L. Althaus"/>
  <meta name="citation_reference" content="citation_title=Estimating epidemic exponential growth rate and basic reproduction number;citation_volume=5;citation_doi=10.1016/j.idm.2019.12.009;citation_issn=2468-0427;citation_author=Junling Ma"/>
  <meta name="citation_reference" content="citation_title=Stochasticity of infectious outbreaks and consequences for optimal interventions;citation_publisher=IOP Publishing;citation_volume=55;citation_doi=10.1088/1751-8121/ac88a6;citation_issn=1751-8121;citation_author=Roberto Morán-Tovar;citation_author=Henning Gruell;citation_author=Florian Klein;citation_author=Michael Lässig"/>
  <!--radix_placeholder_rmarkdown_metadata-->

<script type="text/json" id="radix-rmarkdown-metadata">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","categories","bibliography","date","draft","output"]}},"value":[{"type":"character","attributes":{},"value":["WWTP basic detection theory"]},{"type":"character","attributes":{},"value":["Notes exploring the factors that determine our ability to perform _basic detection_ in a WWTP setting. \nBy basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth."]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url"]}},"value":[{"type":"character","attributes":{},"value":["Michael R. McLaren"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":[]}},"value":[]}]}]},{"type":"character","attributes":{},"value":["theory"]},{"type":"character","attributes":{},"value":["../../_references.bib"]},{"type":"character","attributes":{},"value":["2023-06-03"]},{"type":"logical","attributes":{},"value":[false]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained","dev","toc","code_folding"]}},"value":[{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["svg"]},{"type":"logical","attributes":{},"value":[true]},{"type":"logical","attributes":{},"value":[true]}]}]}]}
</script>
<!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["basic-detection-theory_files/anchor-4.2.2/anchor.min.js","basic-detection-theory_files/bowser-1.9.3/bowser.min.js","basic-detection-theory_files/distill-2.2.21/template.v2.js","basic-detection-theory_files/figure-html5/unnamed-chunk-10-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-12-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-13-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-16-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-17-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-18-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-19-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-2-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-21-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-22-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-23-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-4-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-5-1.svg","basic-detection-theory_files/figure-html5/unnamed-chunk-6-1.svg","basic-detection-theory_files/header-attrs-2.18/header-attrs.js","basic-detection-theory_files/jquery-3.6.0/jquery-3.6.0.js","basic-detection-theory_files/jquery-3.6.0/jquery-3.6.0.min.js","basic-detection-theory_files/jquery-3.6.0/jquery-3.6.0.min.map","basic-detection-theory_files/popper-2.6.0/popper.min.js","basic-detection-theory_files/tippy-6.2.7/tippy-bundle.umd.min.js","basic-detection-theory_files/tippy-6.2.7/tippy-light-border.css","basic-detection-theory_files/tippy-6.2.7/tippy.css","basic-detection-theory_files/tippy-6.2.7/tippy.umd.min.js","basic-detection-theory_files/webcomponents-2.0.0/webcomponents.js"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
<meta name="distill:offset" content="../.."/>

<script type="application/javascript">

  window.headroom_prevent_pin = false;

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // initialize headroom for banner
    var header = $('header').get(0);
    var headerHeight = header.offsetHeight;
    var headroom = new Headroom(header, {
      tolerance: 5,
      onPin : function() {
        if (window.headroom_prevent_pin) {
          window.headroom_prevent_pin = false;
          headroom.unpin();
        }
      }
    });
    headroom.init();
    if(window.location.hash)
      headroom.unpin();
    $(header).addClass('headroom--transition');

    // offset scroll location for banner on hash change
    // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
    window.addEventListener("hashchange", function(event) {
      window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
    });

    // responsive menu
    $('.distill-site-header').each(function(i, val) {
      var topnav = $(this);
      var toggle = topnav.find('.nav-toggle');
      toggle.on('click', function() {
        topnav.toggleClass('responsive');
      });
    });

    // nav dropdowns
    $('.nav-dropbtn').click(function(e) {
      $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
      $(this).parent().siblings('.nav-dropdown')
         .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $("body").click(function(e){
      $('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $(".nav-dropdown").click(function(e){
      e.stopPropagation();
    });
  });
</script>

<style type="text/css">

/* Theme (user-documented overrideables for nav appearance) */

.distill-site-nav {
  color: rgba(255, 255, 255, 0.8);
  background-color: #0F2E3D;
  font-size: 15px;
  font-weight: 300;
}

.distill-site-nav a {
  color: inherit;
  text-decoration: none;
}

.distill-site-nav a:hover {
  color: white;
}

@media print {
  .distill-site-nav {
    display: none;
  }
}

.distill-site-header {

}

.distill-site-footer {

}


/* Site Header */

.distill-site-header {
  width: 100%;
  box-sizing: border-box;
  z-index: 3;
}

.distill-site-header .nav-left {
  display: inline-block;
  margin-left: 8px;
}

@media screen and (max-width: 768px) {
  .distill-site-header .nav-left {
    margin-left: 0;
  }
}


.distill-site-header .nav-right {
  float: right;
  margin-right: 8px;
}

.distill-site-header a,
.distill-site-header .title {
  display: inline-block;
  text-align: center;
  padding: 14px 10px 14px 10px;
}

.distill-site-header .title {
  font-size: 18px;
  min-width: 150px;
}

.distill-site-header .logo {
  padding: 0;
}

.distill-site-header .logo img {
  display: none;
  max-height: 20px;
  width: auto;
  margin-bottom: -4px;
}

.distill-site-header .nav-image img {
  max-height: 18px;
  width: auto;
  display: inline-block;
  margin-bottom: -3px;
}



@media screen and (min-width: 1000px) {
  .distill-site-header .logo img {
    display: inline-block;
  }
  .distill-site-header .nav-left {
    margin-left: 20px;
  }
  .distill-site-header .nav-right {
    margin-right: 20px;
  }
  .distill-site-header .title {
    padding-left: 12px;
  }
}


.distill-site-header .nav-toggle {
  display: none;
}

.nav-dropdown {
  display: inline-block;
  position: relative;
}

.nav-dropdown .nav-dropbtn {
  border: none;
  outline: none;
  color: rgba(255, 255, 255, 0.8);
  padding: 16px 10px;
  background-color: transparent;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  margin: 0;
  margin-top: 1px;
  z-index: 2;
}

.nav-dropdown-content {
  display: none;
  position: absolute;
  background-color: white;
  min-width: 200px;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
  z-index: 1;
  margin-top: 2px;
  white-space: nowrap;
  padding-top: 4px;
  padding-bottom: 4px;
}

.nav-dropdown-content hr {
  margin-top: 4px;
  margin-bottom: 4px;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.nav-dropdown-active {
  display: block;
}

.nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
  color: black;
  padding: 6px 24px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.nav-dropdown-content .nav-dropdown-header {
  display: block;
  padding: 5px 24px;
  padding-bottom: 0;
  text-transform: uppercase;
  font-size: 14px;
  color: #999999;
  white-space: nowrap;
}

.nav-dropdown:hover .nav-dropbtn {
  color: white;
}

.nav-dropdown-content a:hover {
  background-color: #ddd;
  color: black;
}

.nav-right .nav-dropdown-content {
  margin-left: -45%;
  right: 0;
}

@media screen and (max-width: 768px) {
  .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
  .distill-site-header a.nav-toggle {
    float: right;
    display: block;
  }
  .distill-site-header .title {
    margin-left: 0;
  }
  .distill-site-header .nav-right {
    margin-right: 0;
  }
  .distill-site-header {
    overflow: hidden;
  }
  .nav-right .nav-dropdown-content {
    margin-left: 0;
  }
}


@media screen and (max-width: 768px) {
  .distill-site-header.responsive {position: relative; min-height: 500px; }
  .distill-site-header.responsive a.nav-toggle {
    position: absolute;
    right: 0;
    top: 0;
  }
  .distill-site-header.responsive a,
  .distill-site-header.responsive .nav-dropdown {
    display: block;
    text-align: left;
  }
  .distill-site-header.responsive .nav-left,
  .distill-site-header.responsive .nav-right {
    width: 100%;
  }
  .distill-site-header.responsive .nav-dropdown {float: none;}
  .distill-site-header.responsive .nav-dropdown-content {position: relative;}
  .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
    display: block;
    width: 100%;
    text-align: left;
  }
}

/* Site Footer */

.distill-site-footer {
  width: 100%;
  overflow: hidden;
  box-sizing: border-box;
  z-index: 3;
  margin-top: 30px;
  padding-top: 30px;
  padding-bottom: 30px;
  text-align: center;
}

/* Headroom */

d-title {
  padding-top: 6rem;
}

@media print {
  d-title {
    padding-top: 4rem;
  }
}

.headroom {
  z-index: 1000;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.headroom--transition {
  transition: all .4s ease-in-out;
}

.headroom--unpinned {
  top: -100px;
}

.headroom--pinned {
  top: 0;
}

/* adjust viewport for navbar height */
/* helps vertically center bootstrap (non-distill) content */
.min-vh-100 {
  min-height: calc(100vh - 100px) !important;
}

</style>

<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"/>
<link href="../../site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"/>
<script src="../../site_libs/headroom-0.9.4/headroom.min.js"></script>
<script src="../../site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
<script src="../../site_libs/fuse-6.4.1/fuse.min.js"></script>

<script type="application/javascript">

function getMeta(metaName) {
  var metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i++) {
    if (metas[i].getAttribute('name') === metaName) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
}

function offsetURL(url) {
  var offset = getMeta('distill:offset');
  return offset ? offset + '/' + url : url;
}

function createFuseIndex() {

  // create fuse index
  var options = {
    keys: [
      { name: 'title', weight: 20 },
      { name: 'categories', weight: 15 },
      { name: 'description', weight: 10 },
      { name: 'contents', weight: 5 },
    ],
    ignoreLocation: true,
    threshold: 0
  };
  var fuse = new window.Fuse([], options);

  // fetch the main search.json
  return fetch(offsetURL('search.json'))
    .then(function(response) {
      if (response.status == 200) {
        return response.json().then(function(json) {
          // index main articles
          json.articles.forEach(function(article) {
            fuse.add(article);
          });
          // download collections and index their articles
          return Promise.all(json.collections.map(function(collection) {
            return fetch(offsetURL(collection)).then(function(response) {
              if (response.status === 200) {
                return response.json().then(function(articles) {
                  articles.forEach(function(article) {
                    fuse.add(article);
                  });
                })
              } else {
                return Promise.reject(
                  new Error('Unexpected status from search index request: ' +
                            response.status)
                );
              }
            });
          })).then(function() {
            return fuse;
          });
        });

      } else {
        return Promise.reject(
          new Error('Unexpected status from search index request: ' +
                      response.status)
        );
      }
    });
}

window.document.addEventListener("DOMContentLoaded", function (event) {

  // get search element (bail if we don't have one)
  var searchEl = window.document.getElementById('distill-search');
  if (!searchEl)
    return;

  createFuseIndex()
    .then(function(fuse) {

      // make search box visible
      searchEl.classList.remove('hidden');

      // initialize autocomplete
      var options = {
        autoselect: true,
        hint: false,
        minLength: 2,
      };
      window.autocomplete(searchEl, options, [{
        source: function(query, callback) {
          const searchOptions = {
            isCaseSensitive: false,
            shouldSort: true,
            minMatchCharLength: 2,
            limit: 10,
          };
          var results = fuse.search(query, searchOptions);
          callback(results
            .map(function(result) { return result.item; })
          );
        },
        templates: {
          suggestion: function(suggestion) {
            var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
              ? `<img src="${offsetURL(suggestion.preview)}"</img>`
              : '';
            var html = `
              <div class="search-item">
                <h3>${suggestion.title}</h3>
                <div class="search-item-description">
                  ${suggestion.description || ''}
                </div>
                <div class="search-item-preview">
                  ${img}
                </div>
              </div>
            `;
            return html;
          }
        }
      }]).on('autocomplete:selected', function(event, suggestion) {
        window.location.href = offsetURL(suggestion.path);
      });
      // remove inline display style on autocompleter (we want to
      // manage responsive display via css)
      $('.algolia-autocomplete').css("display", "");
    })
    .catch(function(error) {
      console.log(error);
    });

});

</script>

<style type="text/css">

.nav-search {
  font-size: x-small;
}

/* Algolioa Autocomplete */

.algolia-autocomplete {
  display: inline-block;
  margin-left: 10px;
  vertical-align: sub;
  background-color: white;
  color: black;
  padding: 6px;
  padding-top: 8px;
  padding-bottom: 0;
  border-radius: 6px;
  border: 1px #0F2E3D solid;
  width: 180px;
}


@media screen and (max-width: 768px) {
  .distill-site-nav .algolia-autocomplete {
    display: none;
    visibility: hidden;
  }
  .distill-site-nav.responsive .algolia-autocomplete {
    display: inline-block;
    visibility: visible;
  }
  .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
    margin-left: 0;
    width: 400px;
    max-height: 400px;
  }
}

.algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
  width: 90%;
  outline: none;
  border: none;
}

.algolia-autocomplete .aa-hint {
  color: #999;
}
.algolia-autocomplete .aa-dropdown-menu {
  width: 550px;
  max-height: 70vh;
  overflow-x: visible;
  overflow-y: scroll;
  padding: 5px;
  margin-top: 3px;
  margin-left: -150px;
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #999;
  border-top: none;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
  cursor: pointer;
  padding: 5px 4px;
  border-bottom: 1px solid #eee;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
  border-bottom: none;
  margin-bottom: 2px;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
  overflow: hidden;
  font-size: 0.8em;
  line-height: 1.4em;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
  font-size: 1rem;
  margin-block-start: 0;
  margin-block-end: 5px;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
  display: inline-block;
  overflow: hidden;
  height: 2.8em;
  width: 80%;
  margin-right: 4%;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
  display: inline-block;
  width: 15%;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
  height: 3em;
  width: auto;
  display: none;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
  display: initial;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
  background-color: #eee;
}
.algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
  font-weight: bold;
  font-style: normal;
}

</style>


<!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

<style type="text/css">

body {
  background-color: white;
}

.pandoc-table {
  width: 100%;
}

.pandoc-table>caption {
  margin-bottom: 10px;
}

.pandoc-table th:not([align]) {
  text-align: left;
}

.pagedtable-footer {
  font-size: 15px;
}

d-byline .byline {
  grid-template-columns: 2fr 2fr;
}

d-byline .byline h3 {
  margin-block-start: 1.5em;
}

d-byline .byline .authors-affiliations h3 {
  margin-block-start: 0.5em;
}

.authors-affiliations .orcid-id {
  width: 16px;
  height:16px;
  margin-left: 4px;
  margin-right: 4px;
  vertical-align: middle;
  padding-bottom: 2px;
}

d-title .dt-tags {
  margin-top: 1em;
  grid-column: text;
}

.dt-tags .dt-tag {
  text-decoration: none;
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0em 0.4em;
  margin-right: 0.5em;
  margin-bottom: 0.4em;
  font-size: 70%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

d-article table.gt_table td,
d-article table.gt_table th {
  border-bottom: none;
  font-size: 100%;
}

.html-widget {
  margin-bottom: 2.0em;
}

.l-screen-inset {
  padding-right: 16px;
}

.l-screen .caption {
  margin-left: 10px;
}

.shaded {
  background: rgb(247, 247, 247);
  padding-top: 20px;
  padding-bottom: 20px;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .html-widget {
  margin-bottom: 0;
  border: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .shaded-content {
  background: white;
}

.text-output {
  margin-top: 0;
  line-height: 1.5em;
}

.hidden {
  display: none !important;
}

d-article {
  padding-top: 2.5rem;
  padding-bottom: 30px;
}

d-appendix {
  padding-top: 30px;
}

d-article>p>img {
  width: 100%;
}

d-article h2 {
  margin: 1rem 0 1.5rem 0;
}

d-article h3 {
  margin-top: 1.5rem;
}

d-article iframe {
  border: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 2.0em;
  width: 100%;
}

/* Tweak code blocks */

d-article div.sourceCode code,
d-article pre code {
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
}

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: auto;
}

d-article div.sourceCode {
  background-color: white;
}

d-article div.sourceCode pre {
  padding-left: 10px;
  font-size: 12px;
  border-left: 2px solid rgba(0,0,0,0.1);
}

d-article pre {
  font-size: 12px;
  color: black;
  background: none;
  margin-top: 0;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

d-article pre a {
  border-bottom: none;
}

d-article pre a:hover {
  border-bottom: none;
  text-decoration: underline;
}

d-article details {
  grid-column: text;
  margin-bottom: 0.8em;
}

@media(min-width: 768px) {

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: visible !important;
}

d-article div.sourceCode pre {
  padding-left: 18px;
  font-size: 14px;
}

d-article pre {
  font-size: 14px;
}

}

figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

/* CSS for d-contents */

.d-contents {
  grid-column: text;
  color: rgba(0,0,0,0.8);
  font-size: 0.9em;
  padding-bottom: 1em;
  margin-bottom: 1em;
  padding-bottom: 0.5em;
  margin-bottom: 1em;
  padding-left: 0.25em;
  justify-self: start;
}

@media(min-width: 1000px) {
  .d-contents.d-contents-float {
    height: 0;
    grid-column-start: 1;
    grid-column-end: 4;
    justify-self: center;
    padding-right: 3em;
    padding-left: 2em;
  }
}

.d-contents nav h3 {
  font-size: 18px;
  margin-top: 0;
  margin-bottom: 1em;
}

.d-contents li {
  list-style-type: none
}

.d-contents nav > ul {
  padding-left: 0;
}

.d-contents ul {
  padding-left: 1em
}

.d-contents nav ul li {
  margin-top: 0.6em;
  margin-bottom: 0.2em;
}

.d-contents nav a {
  font-size: 13px;
  border-bottom: none;
  text-decoration: none
  color: rgba(0, 0, 0, 0.8);
}

.d-contents nav a:hover {
  text-decoration: underline solid rgba(0, 0, 0, 0.6)
}

.d-contents nav > ul > li > a {
  font-weight: 600;
}

.d-contents nav > ul > li > ul {
  font-weight: inherit;
}

.d-contents nav > ul > li > ul > li {
  margin-top: 0.2em;
}


.d-contents nav ul {
  margin-top: 0;
  margin-bottom: 0.25em;
}

.d-article-with-toc h2:nth-child(2) {
  margin-top: 0;
}


/* Figure */

.figure {
  position: relative;
  margin-bottom: 2.5em;
  margin-top: 1.5em;
}

.figure .caption {
  color: rgba(0, 0, 0, 0.6);
  font-size: 12px;
  line-height: 1.5em;
}

.figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

.figure .caption a {
  color: rgba(0, 0, 0, 0.6);
}

.figure .caption b,
.figure .caption strong, {
  font-weight: 600;
  color: rgba(0, 0, 0, 1.0);
}

/* Citations */

d-article .citation {
  color: inherit;
  cursor: inherit;
}

div.hanging-indent{
  margin-left: 1em; text-indent: -1em;
}

/* Citation hover box */

.tippy-box[data-theme~=light-border] {
  background-color: rgba(250, 250, 250, 0.95);
}

.tippy-content > p {
  margin-bottom: 0;
  padding: 2px;
}


/* Tweak 1000px media break to show more text */

@media(min-width: 1000px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 16px;
  }

  .grid {
    grid-column-gap: 16px;
  }

  d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
  }
  figure .caption, .figure .caption, figure figcaption {
    font-size: 13px;
  }
}

@media(min-width: 1180px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 32px;
  }

  .grid {
    grid-column-gap: 32px;
  }
}


/* Get the citation styles for the appendix (not auto-injected on render since
   we do our own rendering of the citation appendix) */

d-appendix .citation-appendix,
.d-appendix .citation-appendix {
  font-size: 11px;
  line-height: 15px;
  border-left: 1px solid rgba(0, 0, 0, 0.1);
  padding-left: 18px;
  border: 1px solid rgba(0,0,0,0.1);
  background: rgba(0, 0, 0, 0.02);
  padding: 10px 18px;
  border-radius: 3px;
  color: rgba(150, 150, 150, 1);
  overflow: hidden;
  margin-top: -12px;
  white-space: pre-wrap;
  word-wrap: break-word;
}

/* Include appendix styles here so they can be overridden */

d-appendix {
  contain: layout style;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-top: 60px;
  margin-bottom: 0;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  color: rgba(0,0,0,0.5);
  padding-top: 60px;
  padding-bottom: 48px;
}

d-appendix h3 {
  grid-column: page-start / text-start;
  font-size: 15px;
  font-weight: 500;
  margin-top: 1em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.65);
}

d-appendix h3 + * {
  margin-top: 1em;
}

d-appendix ol {
  padding: 0 0 0 15px;
}

@media (min-width: 768px) {
  d-appendix ol {
    padding: 0 0 0 30px;
    margin-left: -30px;
  }
}

d-appendix li {
  margin-bottom: 1em;
}

d-appendix a {
  color: rgba(0, 0, 0, 0.6);
}

d-appendix > * {
  grid-column: text;
}

d-appendix > d-footnote-list,
d-appendix > d-citation-list,
d-appendix > distill-appendix {
  grid-column: screen;
}

/* Include footnote styles here so they can be overridden */

d-footnote-list {
  contain: layout style;
}

d-footnote-list > * {
  grid-column: text;
}

d-footnote-list a.footnote-backlink {
  color: rgba(0,0,0,0.3);
  padding-left: 0.5em;
}



/* Anchor.js */

.anchorjs-link {
  /*transition: all .25s linear; */
  text-decoration: none;
  border-bottom: none;
}
*:hover > .anchorjs-link {
  margin-left: -1.125em !important;
  text-decoration: none;
  border-bottom: none;
}

/* Social footer */

.social_footer {
  margin-top: 30px;
  margin-bottom: 0;
  color: rgba(0,0,0,0.67);
}

.disqus-comments {
  margin-right: 30px;
}

.disqus-comment-count {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  cursor: pointer;
}

#disqus_thread {
  margin-top: 30px;
}

.article-sharing a {
  border-bottom: none;
  margin-right: 8px;
}

.article-sharing a:hover {
  border-bottom: none;
}

.sidebar-section.subscribe {
  font-size: 12px;
  line-height: 1.6em;
}

.subscribe p {
  margin-bottom: 0.5em;
}


.article-footer .subscribe {
  font-size: 15px;
  margin-top: 45px;
}


.sidebar-section.custom {
  font-size: 12px;
  line-height: 1.6em;
}

.custom p {
  margin-bottom: 0.5em;
}

/* Styles for listing layout (hide title) */
.layout-listing d-title, .layout-listing .d-title {
  display: none;
}

/* Styles for posts lists (not auto-injected) */


.posts-with-sidebar {
  padding-left: 45px;
  padding-right: 45px;
}

.posts-list .description h2,
.posts-list .description p {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
}

.posts-list .description h2 {
  font-weight: 700;
  border-bottom: none;
  padding-bottom: 0;
}

.posts-list h2.post-tag {
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  padding-bottom: 12px;
}
.posts-list {
  margin-top: 60px;
  margin-bottom: 24px;
}

.posts-list .post-preview {
  text-decoration: none;
  overflow: hidden;
  display: block;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 24px 0;
}

.post-preview-last {
  border-bottom: none !important;
}

.posts-list .posts-list-caption {
  grid-column: screen;
  font-weight: 400;
}

.posts-list .post-preview h2 {
  margin: 0 0 6px 0;
  line-height: 1.2em;
  font-style: normal;
  font-size: 24px;
}

.posts-list .post-preview p {
  margin: 0 0 12px 0;
  line-height: 1.4em;
  font-size: 16px;
}

.posts-list .post-preview .thumbnail {
  box-sizing: border-box;
  margin-bottom: 24px;
  position: relative;
  max-width: 500px;
}
.posts-list .post-preview img {
  width: 100%;
  display: block;
}

.posts-list .metadata {
  font-size: 12px;
  line-height: 1.4em;
  margin-bottom: 18px;
}

.posts-list .metadata > * {
  display: inline-block;
}

.posts-list .metadata .publishedDate {
  margin-right: 2em;
}

.posts-list .metadata .dt-authors {
  display: block;
  margin-top: 0.3em;
  margin-right: 2em;
}

.posts-list .dt-tags {
  display: block;
  line-height: 1em;
}

.posts-list .dt-tags .dt-tag {
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0.3em 0.4em;
  margin-right: 0.2em;
  margin-bottom: 0.4em;
  font-size: 60%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

.posts-list img {
  opacity: 1;
}

.posts-list img[data-src] {
  opacity: 0;
}

.posts-more {
  clear: both;
}


.posts-sidebar {
  font-size: 16px;
}

.posts-sidebar h3 {
  font-size: 16px;
  margin-top: 0;
  margin-bottom: 0.5em;
  font-weight: 400;
  text-transform: uppercase;
}

.sidebar-section {
  margin-bottom: 30px;
}

.categories ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.categories li {
  color: rgba(0, 0, 0, 0.8);
  margin-bottom: 0;
}

.categories li>a {
  border-bottom: none;
}

.categories li>a:hover {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
}

.categories .active {
  font-weight: 600;
}

.categories .category-count {
  color: rgba(0, 0, 0, 0.4);
}


@media(min-width: 768px) {
  .posts-list .post-preview h2 {
    font-size: 26px;
  }
  .posts-list .post-preview .thumbnail {
    float: right;
    width: 30%;
    margin-bottom: 0;
  }
  .posts-list .post-preview .description {
    float: left;
    width: 45%;
  }
  .posts-list .post-preview .metadata {
    float: left;
    width: 20%;
    margin-top: 8px;
  }
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.5em;
    font-size: 16px;
  }
  .posts-with-sidebar .posts-list {
    float: left;
    width: 75%;
  }
  .posts-with-sidebar .posts-sidebar {
    float: right;
    width: 20%;
    margin-top: 60px;
    padding-top: 24px;
    padding-bottom: 24px;
  }
}


/* Improve display for browsers without grid (IE/Edge <= 15) */

.downlevel {
  line-height: 1.6em;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  margin: 0;
}

.downlevel .d-title {
  padding-top: 6rem;
  padding-bottom: 1.5rem;
}

.downlevel .d-title h1 {
  font-size: 50px;
  font-weight: 700;
  line-height: 1.1em;
  margin: 0 0 0.5rem;
}

.downlevel .d-title p {
  font-weight: 300;
  font-size: 1.2rem;
  line-height: 1.55em;
  margin-top: 0;
}

.downlevel .d-byline {
  padding-top: 0.8em;
  padding-bottom: 0.8em;
  font-size: 0.8rem;
  line-height: 1.8em;
}

.downlevel .section-separator {
  border: none;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
}

.downlevel .d-article {
  font-size: 1.06rem;
  line-height: 1.7em;
  padding-top: 1rem;
  padding-bottom: 2rem;
}


.downlevel .d-appendix {
  padding-left: 0;
  padding-right: 0;
  max-width: none;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.5);
  padding-top: 40px;
  padding-bottom: 48px;
}

.downlevel .footnotes ol {
  padding-left: 13px;
}

.downlevel .base-grid,
.downlevel .distill-header,
.downlevel .d-title,
.downlevel .d-abstract,
.downlevel .d-article,
.downlevel .d-appendix,
.downlevel .distill-appendix,
.downlevel .d-byline,
.downlevel .d-footnote-list,
.downlevel .d-citation-list,
.downlevel .distill-footer,
.downlevel .appendix-bottom,
.downlevel .posts-container {
  padding-left: 40px;
  padding-right: 40px;
}

@media(min-width: 768px) {
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
  padding-left: 150px;
  padding-right: 150px;
  max-width: 900px;
}
}

.downlevel pre code {
  display: block;
  border-left: 2px solid rgba(0, 0, 0, .1);
  padding: 0 0 0 20px;
  font-size: 14px;
}

.downlevel code, .downlevel pre {
  color: black;
  background: none;
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.downlevel .posts-list .post-preview {
  color: inherit;
}



</style>

<script type="application/javascript">

function is_downlevel_browser() {
  if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                 window.navigator.userAgent)) {
    return true;
  } else {
    return window.load_distill_framework === undefined;
  }
}

// show body when load is complete
function on_load_complete() {

  // add anchors
  if (window.anchors) {
    window.anchors.options.placement = 'left';
    window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
  }


  // set body to visible
  document.body.style.visibility = 'visible';

  // force redraw for leaflet widgets
  if (window.HTMLWidgets) {
    var maps = window.HTMLWidgets.findAll(".leaflet");
    $.each(maps, function(i, el) {
      var map = this.getMap();
      map.invalidateSize();
      map.eachLayer(function(layer) {
        if (layer instanceof L.TileLayer)
          layer.redraw();
      });
    });
  }

  // trigger 'shown' so htmlwidgets resize
  $('d-article').trigger('shown');
}

function init_distill() {

  init_common();

  // create front matter
  var front_matter = $('<d-front-matter></d-front-matter>');
  $('#distill-front-matter').wrap(front_matter);

  // create d-title
  $('.d-title').changeElementType('d-title');

  // create d-byline
  var byline = $('<d-byline></d-byline>');
  $('.d-byline').replaceWith(byline);

  // create d-article
  var article = $('<d-article></d-article>');
  $('.d-article').wrap(article).children().unwrap();

  // move posts container into article
  $('.posts-container').appendTo($('d-article'));

  // create d-appendix
  $('.d-appendix').changeElementType('d-appendix');

  // flag indicating that we have appendix items
  var appendix = $('.appendix-bottom').children('h3').length > 0;

  // replace footnotes with <d-footnote>
  $('.footnote-ref').each(function(i, val) {
    appendix = true;
    var href = $(this).attr('href');
    var id = href.replace('#', '');
    var fn = $('#' + id);
    var fn_p = $('#' + id + '>p');
    fn_p.find('.footnote-back').remove();
    var text = fn_p.html();
    var dtfn = $('<d-footnote></d-footnote>');
    dtfn.html(text);
    $(this).replaceWith(dtfn);
  });
  // remove footnotes
  $('.footnotes').remove();

  // move refs into #references-listing
  $('#references-listing').replaceWith($('#refs'));

  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    var id = $(this).attr('id');
    $('.d-contents a[href="#' + id + '"]').parent().remove();
    appendix = true;
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
  });

  // show d-appendix if we have appendix content
  $("d-appendix").css('display', appendix ? 'grid' : 'none');

  // localize layout chunks to just output
  $('.layout-chunk').each(function(i, val) {

    // capture layout
    var layout = $(this).attr('data-layout');

    // apply layout to markdown level block elements
    var elements = $(this).children().not('details, div.sourceCode, pre, script');
    elements.each(function(i, el) {
      var layout_div = $('<div class="' + layout + '"></div>');
      if (layout_div.hasClass('shaded')) {
        var shaded_content = $('<div class="shaded-content"></div>');
        $(this).wrap(shaded_content);
        $(this).parent().wrap(layout_div);
      } else {
        $(this).wrap(layout_div);
      }
    });


    // unwrap the layout-chunk div
    $(this).children().unwrap();
  });

  // remove code block used to force  highlighting css
  $('.distill-force-highlighting-css').parent().remove();

  // remove empty line numbers inserted by pandoc when using a
  // custom syntax highlighting theme
  $('code.sourceCode a:empty').remove();

  // load distill framework
  load_distill_framework();

  // wait for window.distillRunlevel == 4 to do post processing
  function distill_post_process() {

    if (!window.distillRunlevel || window.distillRunlevel < 4)
      return;

    // hide author/affiliations entirely if we have no authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
    if (!have_authors)
      $('d-byline').addClass('hidden');

    // article with toc class
    $('.d-contents').parent().addClass('d-article-with-toc');

    // strip links that point to #
    $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

    // add orcid ids
    $('.authors-affiliations').find('.author').each(function(i, el) {
      var orcid_id = front_matter.authors[i].orcidID;
      if (orcid_id) {
        var a = $('<a></a>');
        a.attr('href', 'https://orcid.org/' + orcid_id);
        var img = $('<img></img>');
        img.addClass('orcid-id');
        img.attr('alt', 'ORCID ID');
        img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
        a.append(img);
        $(this).append(a);
      }
    });

    // hide elements of author/affiliations grid that have no value
    function hide_byline_column(caption) {
      $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
    }

    // affiliations
    var have_affiliations = false;
    for (var i = 0; i<front_matter.authors.length; ++i) {
      var author = front_matter.authors[i];
      if (author.affiliation !== "&nbsp;") {
        have_affiliations = true;
        break;
      }
    }
    if (!have_affiliations)
      $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

    // published date
    if (!front_matter.publishedDate)
      hide_byline_column("Published");

    // document object identifier
    var doi = $('d-byline').find('h3:contains("DOI")');
    var doi_p = doi.next().empty();
    if (!front_matter.doi) {
      // if we have a citation and valid citationText then link to that
      if ($('#citation').length > 0 && front_matter.citationText) {
        doi.html('Citation');
        $('<a href="#citation"></a>')
          .text(front_matter.citationText)
          .appendTo(doi_p);
      } else {
        hide_byline_column("DOI");
      }
    } else {
      $('<a></a>')
         .attr('href', "https://doi.org/" + front_matter.doi)
         .html(front_matter.doi)
         .appendTo(doi_p);
    }

     // change plural form of authors/affiliations
    if (front_matter.authors.length === 1) {
      var grid = $('.authors-affiliations');
      grid.children('h3:contains("Authors")').text('Author');
      grid.children('h3:contains("Affiliations")').text('Affiliation');
    }

    // remove d-appendix and d-footnote-list local styles
    $('d-appendix > style:first-child').remove();
    $('d-footnote-list > style:first-child').remove();

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // hoverable references
    $('span.citation[data-cites]').each(function() {
      const citeChild = $(this).children()[0]
      // Do not process if @xyz has been used without escaping and without bibliography activated
      // https://github.com/rstudio/distill/issues/466
      if (citeChild === undefined) return true

      if (citeChild.nodeName == "D-FOOTNOTE") {
        var fn = citeChild
        $(this).html(fn.shadowRoot.querySelector("sup"))
        $(this).id = fn.id
        fn.remove()
      }
      var refs = $(this).attr('data-cites').split(" ");
      var refHtml = refs.map(function(ref) {
        // Could use CSS.escape too here, we insure backward compatibility in navigator
        return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
      }).join("\n");
      window.tippy(this, {
        allowHTML: true,
        content: refHtml,
        maxWidth: 500,
        interactive: true,
        interactiveBorder: 10,
        theme: 'light-border',
        placement: 'bottom-start'
      });
    });

    // fix footnotes in tables (#411)
    // replacing broken distill.pub feature
    $('table d-footnote').each(function() {
      // we replace internal showAtNode methode which is triggered when hovering a footnote
      this.hoverBox.showAtNode = function(node) {
        // ported from https://github.com/distillpub/template/pull/105/files
        calcOffset = function(elem) {
            let x = elem.offsetLeft;
            let y = elem.offsetTop;
            // Traverse upwards until an `absolute` element is found or `elem`
            // becomes null.
            while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                x += elem.offsetLeft;
                y += elem.offsetTop;
            }

            return { left: x, top: y };
        }
        // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
        const bbox = node.getBoundingClientRect();
        const offset = calcOffset(node);
        this.show([offset.left + bbox.width, offset.top + bbox.height]);
      }
    })

    // clear polling timer
    clearInterval(tid);

    // show body now that everything is ready
    on_load_complete();
  }

  var tid = setInterval(distill_post_process, 50);
  distill_post_process();

}

function init_downlevel() {

  init_common();

   // insert hr after d-title
  $('.d-title').after($('<hr class="section-separator"/>'));

  // check if we have authors
  var front_matter = JSON.parse($("#distill-front-matter").html());
  var have_authors = front_matter.authors && front_matter.authors.length > 0;

  // manage byline/border
  if (!have_authors)
    $('.d-byline').remove();
  $('.d-byline').after($('<hr class="section-separator"/>'));
  $('.d-byline a').remove();

  // remove toc
  $('.d-contents').remove();

  // move appendix elements
  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
  });


  // inject headers into references and footnotes
  var refs_header = $('<h3></h3>');
  refs_header.text('References');
  $('#refs').prepend(refs_header);

  var footnotes_header = $('<h3></h3');
  footnotes_header.text('Footnotes');
  $('.footnotes').children('hr').first().replaceWith(footnotes_header);

  // move appendix-bottom entries to the bottom
  $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
  $('.appendix-bottom').remove();

  // remove appendix if it's empty
  if ($('.d-appendix').children().length === 0)
    $('.d-appendix').remove();

  // prepend separator above appendix
  $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

  // trim code
  $('pre>code').each(function(i, val) {
    $(this).html($.trim($(this).html()));
  });

  // move posts-container right before article
  $('.posts-container').insertBefore($('.d-article'));

  $('body').addClass('downlevel');

  on_load_complete();
}


function init_common() {

  // jquery plugin to change element types
  (function($) {
    $.fn.changeElementType = function(newType) {
      var attrs = {};

      $.each(this[0].attributes, function(idx, attr) {
        attrs[attr.nodeName] = attr.nodeValue;
      });

      this.replaceWith(function() {
        return $("<" + newType + "/>", attrs).append($(this).contents());
      });
    };
  })(jQuery);

  // prevent underline for linked images
  $('a > img').parent().css({'border-bottom' : 'none'});

  // mark non-body figures created by knitr chunks as 100% width
  $('.layout-chunk').each(function(i, val) {
    var figures = $(this).find('img, .html-widget');
    // ignore leaflet img layers (#106)
    figures = figures.filter(':not(img[class*="leaflet"])')
    if ($(this).attr('data-layout') !== "l-body") {
      figures.css('width', '100%');
    } else {
      figures.css('max-width', '100%');
      figures.filter("[width]").each(function(i, val) {
        var fig = $(this);
        fig.css('width', fig.attr('width') + 'px');
      });

    }
  });

  // auto-append index.html to post-preview links in file: protocol
  // and in rstudio ide preview
  $('.post-preview').each(function(i, val) {
    if (window.location.protocol === "file:")
      $(this).attr('href', $(this).attr('href') + "index.html");
  });

  // get rid of index.html references in header
  if (window.location.protocol !== "file:") {
    $('.distill-site-header a[href]').each(function(i,val) {
      $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
    });
  }

  // add class to pandoc style tables
  $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
  $('.kable-table').children('table').addClass('pandoc-table');

  // add figcaption style to table captions
  $('caption').parent('table').addClass("figcaption");

  // initialize posts list
  if (window.init_posts_list)
    window.init_posts_list();

  // implmement disqus comment link
  $('.disqus-comment-count').click(function() {
    window.headroom_prevent_pin = true;
    $('#disqus_thread').toggleClass('hidden');
    if (!$('#disqus_thread').hasClass('hidden')) {
      var offset = $(this).offset();
      $(window).resize();
      $('html, body').animate({
        scrollTop: offset.top - 35
      });
    }
  });
}

document.addEventListener('DOMContentLoaded', function() {
  if (is_downlevel_browser())
    init_downlevel();
  else
    window.addEventListener('WebComponentsReady', init_distill);
});

</script>

<!--/radix_placeholder_distill-->
  <script src="../../site_libs/header-attrs-2.18/header-attrs.js"></script>
  <script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <script src="../../site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="../../site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="../../site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="../../site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="../../site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="../../site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="../../site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="../../site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
<!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"WWTP basic detection theory","description":"Notes exploring the factors that determine our ability to perform _basic detection_ in a WWTP setting. \nBy basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth.","authors":[{"author":"Michael R. McLaren","authorURL":{},"affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2023-06-03T00:00:00.000+00:00","citationText":"McLaren, 2023"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="../../index.html" class="title">Mike's NAO notebook</a>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="../../index.html">Home</a>
<a href="../../about.html">About</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>WWTP basic detection theory</h1>
<!--radix_placeholder_categories-->
<div class="dt-tags">
  <a href="../../index.html#category:theory" class="dt-tag">theory</a>
</div>
<!--/radix_placeholder_categories-->
<p><p>Notes exploring the factors that determine our ability to perform <em>basic detection</em> in a WWTP setting.
By basic detection, I mean detection based on seeing a sufficient number of distinguishing reads from the pathogen, rather than from a spatiotemporal pattern such as exponential growth.</p></p>
</div>

<div class="d-byline">
  Michael R. McLaren true 
  
<br/>2023-06-03
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#detection-under-deterministic-exponential-growth" id="toc-detection-under-deterministic-exponential-growth">Detection under deterministic exponential growth</a>
<ul>
<li><a href="#epidemic-dynamics" id="toc-epidemic-dynamics">Epidemic dynamics</a></li>
<li><a href="#shedding-and-sequencing" id="toc-shedding-and-sequencing">Shedding and sequencing</a></li>
<li><a href="#detection" id="toc-detection">Detection</a>
<ul>
<li><a href="#cumulative-reads-over-a-given-period" id="toc-cumulative-reads-over-a-given-period">Cumulative reads over a given period</a></li>
</ul></li>
<li><a href="#further-discussion" id="toc-further-discussion">Further discussion</a></li>
</ul></li>
<li><a href="#applications" id="toc-applications">Applications</a>
<ul>
<li><a href="#example-sars2-reads-in-southern-california-wwtp-data" id="toc-example-sars2-reads-in-southern-california-wwtp-data">Example: SARS2 reads in Southern California WWTP data</a></li>
</ul></li>
<li><a href="#details-and-complications" id="toc-details-and-complications">Details and complications</a>
<ul>
<li><a href="#complex-disease-time-courses" id="toc-complex-disease-time-courses">Complex disease time courses</a></li>
<li><a href="#seir-model" id="toc-seir-model">SEIR model</a></li>
<li><a href="#random-variation-in-transmission" id="toc-random-variation-in-transmission">(Random) variation in transmission</a>
<ul>
<li><a href="#relevant-references" id="toc-relevant-references">Relevant references</a></li>
</ul></li>
<li><a href="#origination-of-the-outbreak" id="toc-origination-of-the-outbreak">Origination of the outbreak</a></li>
<li><a href="#shedding" id="toc-shedding">Shedding</a></li>
<li><a href="#transport" id="toc-transport">Transport</a>
<ul>
<li><a href="#effect-of-transport-time" id="toc-effect-of-transport-time">Effect of transport time</a></li>
<li><a href="#changes-in-waste-composition-during-transport" id="toc-changes-in-waste-composition-during-transport">Changes in waste composition during transport</a></li>
</ul></li>
<li><a href="#sample-collection" id="toc-sample-collection">Sample collection</a>
<ul>
<li><a href="#partitioning-into-solid-and-liquid-fractions" id="toc-partitioning-into-solid-and-liquid-fractions">Partitioning into solid and liquid fractions</a></li>
</ul></li>
<li><a href="#detection-methods" id="toc-detection-methods">Detection methods</a>
<ul>
<li><a href="#metagenomic-sequencing" id="toc-metagenomic-sequencing">Metagenomic sequencing</a></li>
<li><a href="#targeted-detection-using-qpcr-or-dpcr" id="toc-targeted-detection-using-qpcr-or-dpcr">Targeted detection using qPCR or dPCR</a></li>
<li><a href="#amplicon-sequencing" id="toc-amplicon-sequencing">Amplicon sequencing</a></li>
</ul></li>
</ul></li>
<li><a href="#notes" id="toc-notes">Notes</a>
<ul>
<li><a href="#transportation-time" id="toc-transportation-time">Transportation time</a></li>
<li><a href="#effective-number-of-shedders-contributing-to-a-wwtp-sample" id="toc-effective-number-of-shedders-contributing-to-a-wwtp-sample">Effective number of shedders contributing to a WWTP sample</a></li>
<li><a href="#non-human-contributions-to-wwtp-samples" id="toc-non-human-contributions-to-wwtp-samples">Non-human contributions to WWTP samples</a></li>
<li><a href="#noisevariability-in-the-model" id="toc-noisevariability-in-the-model">Noise/variability in the model</a></li>
</ul></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://fs.r-lib.org'>fs</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># plotting helpers</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/eclarke/ggbeeswarm'>ggbeeswarm</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme_get.html'>theme_set</a></span><span class='op'>(</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># Okabe Ito color scheme with amber for yellow; see https://easystats.github.io/see/reference/scale_color_okabeito.html</span></span>
<span><span class='va'>colors_oi</span> <span class='op'>&lt;-</span> <span class='fu'>grDevices</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/r/grDevices/palette.html'>palette.colors</a></span><span class='op'>(</span><span class='op'>)</span>  </span>
<span><span class='va'>colors_oi</span><span class='op'>[</span><span class='st'>'yellow'</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='st'>"#F5C710"</span></span></code></pre>
</div>
</details>
</div>
<!-- Links

- [Initial Overleaf modeling doc](https://www.overleaf.com/project/62ea87d86acf9339a72ed433)
- [Threat portfolio report](https://docs.google.com/document/d/1_qjSuCUBAKBBVV04FzEdt4XYZB6qZpCu1hs27YdVfLs/)

-->
<h1 id="detection-under-deterministic-exponential-growth">Detection under deterministic exponential growth</h1>
<p>I will assume a very simple model, but which can represent more complex cases so long as the outbreak has has established, and reached the asymptotic exponential growth rate in the number of infections, cumulative infections, and shedding amount, by using ‘effective parameters’.
Motivation: Mathematical simplicity for developing intuition; also, I assume that detection has a negligible chance of occurring before the outbreak reaches the exponential phase.</p>
<p>Currently considers MGS-based detection only; future versions may add targeted detection and amplicon sequencing.</p>
<p>Description of the basic model:</p>
<ul>
<li>Deterministic exponential growth in the number of infected individuals.
<ul>
<li>Intended to model the early stage of an outbreak where less than <span class="math inline">\(\sim 1/3\)</span> of the population have been infected.</li>
</ul></li>
<li>The ratios of infected to infectious and to the total shedding rate assumed constant.
<ul>
<li>Individual variation and randomness in disease time course and shedding are treated as affecting these ratios, and otherwise ignored.</li>
</ul></li>
<li>Microbiome background shedding assumed constant and independent of infection status.</li>
<li>MGS results only depend on the relative abundances in the sample.</li>
<li>Effect of WW transport, sample collection, and sample processing on relative abundance of pathogen to background can be represented by a constant bias parameter.</li>
<li>Variability/noise in MGS read counts due to transport, sample collection, processing, and sequencing is ignored.</li>
</ul>
<p>In this section, we will use this basic model to derive the time, current infected fraction, and cumulative incidence when <span class="math inline">\(M^*\)</span> reads of a pathogen (or a particular k-mer in the pathogen’s genome) are first seen on a single day or cumulatively.</p>
<p>TODO</p>
<ul class="task-list">
<li><input type="checkbox" />above, better distinguish between motivation/justification of the model and statement of the model.</li>
<li><input type="checkbox" />above, better describe the purpose/message of this section</li>
<li><input type="checkbox" />below, review and consider which details to move to later sections or an appendix</li>
</ul>
<h2 id="epidemic-dynamics">Epidemic dynamics</h2>
<p>Notation warning: Here, I’m taking <span class="math inline">\(I\)</span> to be the number of infections, which includes exposed but non-yet-infectious individuals.
In other words, this would be <span class="math inline">\(E + I\)</span> in the SEIR model.</p>
<p>Assume deterministic exponential growth of the current number of infected, <span class="math inline">\(I(t)\)</span>, from some initial number <span class="math inline">\(I_0 = I(0)\)</span>.
The number of infected at time <span class="math inline">\(t\)</span> (measured in days) has derivative
<span class="math display">\[\begin{align}
  \frac{dI}{dt} &amp;= (b - d) I = r I,
\end{align}\]</span>
where <span class="math inline">\(r &gt; 0\)</span> is the exponential growth rate.
I further assume that <span class="math inline">\(r\)</span> can be broken into separate ‘birth’ and ‘death’ rates, <span class="math inline">\(b\)</span> and <span class="math inline">\(d\)</span>, describing the rate at which infections beget new infections, and which infected individuals recover.
Under this model, the number of infections at time <span class="math inline">\(t\)</span> grows as
<span class="math display">\[\begin{align}
  I(t) &amp;\approx I_0 e^{rt}.
\end{align}\]</span></p>
<p>The cumulative number of infection-days at day <span class="math inline">\(t\)</span> is
<span class="math display">\[\begin{align}
  \int_0^t I(u)\;du
    &amp;= \int_0^t I_0 e^{ru}\;du
  \\&amp;= \frac{I_0}{r} \left(e^{rt} - 1 \right)
  \\&amp;= \frac{1}{r} \left[I(t) - I(0) \right]
  \\&amp;\approx \frac{1}{r} I(t) \quad \text{for $t \gg 1/r$}.
\end{align}\]</span></p>
<p>We can calculate the cumulative number of infections <span class="math inline">\(C(t)\)</span> at day <span class="math inline">\(t\)</span> by noting that new infections occur at rate <span class="math inline">\(b I(t)\)</span>; hence
<span class="math display">\[\begin{align}
  C(t)
    &amp;= \int_0^t b I(u)\;du
  \\&amp;= \int_0^t b I_0 e^{ru}\;du
  \\&amp;= \frac{b I_0}{r} \left(e^{rt} - 1 \right)
  \\&amp;= \frac{b}{r} \left[I(t) - I(0) \right]
  \\&amp;\approx \frac{b}{r} I(t) \quad \text{for $t \gg 1/r$}.
\end{align}\]</span>
The cumulative number of infections is <span class="math inline">\(b\)</span> times the cumulative number of infection-days, and is approximately <span class="math inline">\(b/r\)</span> the number of current infections.
Note: Arguably, we should be adding the initial number of infected, <span class="math inline">\(I_0\)</span>, in which case we get
<span class="math display">\[\begin{align}
  C(t)
    &amp;= \int_0^t b I(u)\;du  + I_0
  \\&amp;= \frac{b}{r} \left[I(t) - I_0 \right] + I_0
  \\&amp;= \frac{b}{r} I(t) - \frac{d}{r} I_0.
  \\&amp;\approx \frac{b}{r} I(t) \quad \text{for $I(t) \gg I_0$}.
\end{align}\]</span>
I will typically assume <span class="math inline">\(I_0\)</span> is a tiny fraction of all current and cumulative infections in the regimes of interest, so this distinction shouldn’t matter.
The condition <span class="math inline">\(I(t) \gg I_0\)</span> will be true for <span class="math inline">\(t \gg 1/r\)</span>.</p>
<p>Define <em>cumulative incidence</em> <span class="math inline">\(c(t)\)</span> as the fraction of the population that have been infected by time <span class="math inline">\(t\)</span>,
<span class="math display">\[\begin{align}
  c(t)
    &amp;= C(t) / N
  \\&amp;\approx \frac{b}{r} i(t) \quad \text{for $t \gg 1/r$}.
\end{align}\]</span></p>
<p>How long is the exponential approximation good for?
Suppose there is lasting immunity, such that each person can only be infected once.
In that case, we expect the rate of new infections per capita, <span class="math inline">\(b\)</span>, to be in proportional to <span class="math inline">\(1 - c(t)\)</span>; i.e., <span class="math inline">\(b(t) = [1 - c(t)] b_0\)</span>.
The exponential approximation is good so long as <span class="math inline">\(c(t) \ll 1\)</span>.
A good rule of thumb could be to take the point when <span class="math inline">\(c(t) \approx 1/e\)</span> as the end of the exponential phase.
This occurs when the currently infected fraction of the population <span class="math inline">\(i(t)\)</span> reaches <span class="math inline">\(\tfrac{r}{b} e^{-1}\)</span>, or when <span class="math inline">\(t \approx \tfrac{1}{r} [\log (\tfrac{r}{i_0 b}) - 1]\)</span>.
(Needs to be checked)</p>
<p>The peak of the epidemic (max value of <span class="math inline">\(I\)</span>) will occur when <span class="math inline">\(b\)</span> decreases to <span class="math inline">\(d\)</span>.
This occurs when <span class="math inline">\(c(t) = 1 - \tfrac{d}{b_0} = \tfrac{r_0}{b_0}\)</span>.
The number of susceptible (those not infected) is <span class="math inline">\(1 - c = d/b_0\)</span>, matching the standard SIR result (e.g. equation 4a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7997702/">here</a>).</p>
<h2 id="shedding-and-sequencing">Shedding and sequencing</h2>
<p>Suppose the number of infected at time <span class="math inline">\(t\)</span> is then
<span class="math display">\[\begin{align}
  I(t) &amp;\approx I_0 e^{rt}.
\end{align}\]</span>
where <span class="math inline">\(I_0\)</span> is the initial number of infected.
Let <span class="math inline">\(N\)</span> be the total population (assumed fixed) and <span class="math inline">\(i(t) = I(t) / N\)</span> be the fraction of the population that is infected.</p>
<p>Assume a constant amount of background shedding in each person regardless of infection status.
Suppose a background shedding rate of <span class="math inline">\(s_0\)</span> of background microbiome in each person, regardless of infection status.
Let <span class="math inline">\(s\)</span> be the relative rate of shedding of pathogen to background in an infected person, so that <span class="math inline">\(s s_0\)</span> is the rate of pathogen shedding in an infected person (note potential name clash with the fraction of the population that is susceptible).
Let <span class="math inline">\(B\)</span> be the relative measurement effiency of the pathogen to the background (MGS bias).
In the regime I consider below, where total threat reads are a small fraction of total reads, we can use <span class="math inline">\(B\)</span> to also account for the fact that perhaps only a small fraction of reads from the pathogen are useful for detecting it as a novel threat.</p>
<p>The expected fraction (i.e. proportion) of the threat in the MGS reads in a sample collected on day <span class="math inline">\(t\)</span> is
<span class="math display">\[\begin{align}
E [ P(t) ]
  &amp;\approx \frac{I(t) s s_0 B}{I(t) s s_0  B + N s_0 }
\\&amp;= \frac{I(t) s B}{I(t) s B + N }
\\&amp;= \frac{i(t) s B}{i(t) s B + 1 }
\\&amp;\approx i(t) s B.
\end{align}\]</span>
The first approximation comes from treating <span class="math inline">\(I(t)\)</span> as constant over the 24h the sample is collected (more on this below).
The second approximation comes from assuming that the pathogen is always a small fraction of the reads (<span class="math inline">\(i s B \ll 1\)</span>).
One way for this condition to arise is that it holds even for a sample from an infected individual or an entirely infected population (i.e., <span class="math inline">\(s B \ll 1\)</span>).
I expect that to be true for subtle pathogens, but not necessary all pathogens (e.g., gastroenteric pathogens).
For these latter cases I expect <span class="math inline">\(i(t) \ll 1\)</span> and for the broader condition <span class="math inline">\(i s B \ll 1\)</span> to still hold.</p>
<p>To determine the absolute copy number or concentration of the pathogen in the sample, we’d need to say something about the overall shedding amount and water system. However, to analyze MGS detection, it is enough to consider the relative abundance as above.</p>
<p>The expected number of sequencing reads of the threat on day <span class="math inline">\(t\)</span> is
<span class="math display">\[\begin{align}
E [ M(t) ] = E [ P(t) ] \cdot \mathcal M \approx i(t) s B \mathcal M.
\end{align}\]</span>
where <span class="math inline">\(\mathcal M\)</span> is the total sequencing depth, which I’m treating as a value that is determined by the experimenter and is the same for each day.</p>
<p>More generally, if the sample is collected over a particular period of time <span class="math inline">\(\Delta t\)</span>, we might want to model the pathogen contribution to the sample as an integral of shedding over that period,
<span class="math display">\[\begin{align}
E [ M(t) ]
  &amp;= s B \mathcal M \int_{t-\Delta t}{t} i(u) \; du
\\&amp;= s B \mathcal M \frac{1}{r} i_0 e^{rt} \left[1 - e^{-r \Delta t} \right]
\\&amp;= s B \mathcal M \frac{1}{r} i(t) \left[1 - e^{-r \Delta t} \right].
\\&amp;\approx i(t) (\Delta t) s B \mathcal M \quad \text{for $r \Delta t \ll 1$}.
\end{align}\]</span></p>
<h2 id="detection">Detection</h2>
<p>Warning: The below assumes that <span class="math inline">\(r \ll 1\)</span>, so that we can ignore the growth in <span class="math inline">\(i(t)\)</span> over the sampling period when computing the contribution of pathogen to the sample.</p>
<p>Suppose there is a threshold number of reads (of the pathogen or the distinguishing subsequence), <span class="math inline">\(M^*\)</span>, we need to see in a single day to detect the threat (e.g. 5).
I’ll take <span class="math inline">\(\Delta t = 1\)</span> (24-h composite sampling) and suppose that <span class="math inline">\(r \ll 1\)</span>, so that we can ignore growth over the sampling window (otherwise, just multiply the number of expected reads by the constant factor <span class="math inline">\((1 - e^r)/r\)</span>).
As a first approximation, let’s ignore noise in shedding and sequencing measurement, and define detection as occurring when <span class="math inline">\(E [M(t^*)] = M^*\)</span>.
Detection then occurs at <span class="math inline">\(i(t) s B \mathcal M = M^*\)</span> or
<span class="math display">\[\begin{align}
  i(t^*) = \frac{M^*}{s B \mathcal M}.
\end{align}\]</span>
In the exponential regime where <span class="math inline">\(i(t) = i_0 e^{rt}\)</span> and cumulative incidence is <span class="math inline">\(c(t) \approx \tfrac{b}{r} i(t)\)</span>, we have that
<span class="math display">\[\begin{align}
  t^* = \frac{1}{r} \log \frac{M^*}{i_0 s B \mathcal M}
\end{align}\]</span>
<span class="math display">\[\begin{align}
  c(t^*) = \frac{b}{r} \cdot \frac{M^*}{s B \mathcal M}.
\end{align}\]</span></p>
<p>Note that the detection threshold and efficiency appear together as <span class="math inline">\(M^* / B\)</span>; this means we’ll get the same answer whether we think of</p>
<ul>
<li><span class="math inline">\(M^*\)</span> is the number of reads hitting a particular identifying subsequence, and <span class="math inline">\(B\)</span> contains a factor <span class="math inline">\(x &lt; 1\)</span> to account for the fact that only a fraction <span class="math inline">\(x\)</span> of the pathogen’s reads hit the subsequence</li>
<li><span class="math inline">\(M^*\)</span> is the number of reads hitting the pathogen, and we increase it by a factor <span class="math inline">\(1/x\)</span> to account for the fact that only <span class="math inline">\(x\)</span> of the reads are useful for identification; <span class="math inline">\(B\)</span> does not contain the factor <span class="math inline">\(x\)</span>.</li>
</ul>
<p>However, when we start to consider noise in the read counts, we need to be more careful about this.</p>
<p><strong>Joint dependence of many factors:</strong>
The effect of <span class="math inline">\(s\)</span>, <span class="math inline">\(B\)</span>, <span class="math inline">\(M^*\)</span> and <span class="math inline">\(\mathcal M\)</span> all occurs through the term <span class="math inline">\(\frac{M^*}{s B \mathcal M}\)</span>.
(This should match our preexisting intuition.)
Thus, for example, we can lower sequencing by 2-fold if we can enrich for the pathogen 2-fold (by, for exampling, doing a better job of removing bacterial NA for viral detection).</p>
<p><strong>Effect of sequencing effort:</strong>
These results explain the pattern seen in the example simulations presented by Charlie Whittaker last December, in which we saw that the time of detection decreases logarithmically <span class="math inline">\(\mathcal M\)</span> and cumulative incidence at time of detection decreases as <span class="math inline">\(1 / \mathcal M\)</span>.
As per the previous point, we see the same behavior when we vary <span class="math inline">\(s\)</span>, <span class="math inline">\(B\)</span>, or <span class="math inline">\(1/M^*\)</span>.</p>
<h3 id="cumulative-reads-over-a-given-period">Cumulative reads over a given period</h3>
<p>What if it is sufficient to see <span class="math inline">\(R^*\)</span> reads cumulatively over the last <span class="math inline">\(T\)</span> days?
Call <span class="math inline">\(Q\)</span> the cumulative reads over the last <span class="math inline">\(T\)</span> days.
Then
<span class="math display">\[\begin{align}
E [ Q(t) ]
  &amp;= s B \mathcal M \int_{t-T}{t} i(u) \; du
\\&amp;= s B \mathcal M i(t) \frac{1}{r} \left[1 - e^{-r T} \right].
\end{align}\]</span>
Setting <span class="math inline">\(T = t\)</span> corresponds to all reads since time <span class="math inline">\(0\)</span>; in that case, we have that <span class="math inline">\(E [ Q(t) ]\)</span> equals <span class="math inline">\(s B \mathcal M\)</span> times the cumulative infection hours, or approximately
<span class="math display">\[\begin{align}
E [ Q(t) ]
  &amp;\approx s B \mathcal M \frac{i(t)}{r}
  \\&amp;\approx s B \mathcal M \frac{c(t)}{b}
\end{align}\]</span>
Considering the factor <span class="math inline">\(1 - e^{-rT}\)</span> indicates that it is only the samples from the past few characteristic growth periods that contributes significantly to the total reads.</p>
<p>How much better do we do, detection wise, when we only require cumulative reads to reach <span class="math inline">\(M^*\)</span>?
Detection now occurs at <span class="math inline">\(E[Q(t)] = M^*\)</span> or
<span class="math display">\[\begin{align}
  i(t^*) &amp;= \frac{r M^*}{s B \mathcal M} \\
  c(t^*) &amp;= b \cdot \frac{M^*}{s B \mathcal M};
\end{align}\]</span>
these are both a factor <span class="math inline">\(r\)</span> greater than before.
As expected, the effect on time is only logarithmic,
<span class="math display">\[\begin{align}
  t^* = \frac{1}{r} \log \frac{r M^*}{i_0 s B \mathcal M}.
\end{align}\]</span>
The effect overall is as if we increased any of <span class="math inline">\(\mathcal M\)</span>, <span class="math inline">\(s\)</span>, or <span class="math inline">\(B\)</span> by a factor of <span class="math inline">\(1/r\)</span>.</p>
<h2 id="further-discussion">Further discussion</h2>
<h4 id="mgs-monitoring-performance-is-independent-of-wwtp-catchment-size-and-flow-rate">MGS monitoring performance is independent of WWTP catchment size and flow rate</h4>
<ul>
<li>WWTP catchment size and flow rate do not affect the quantities of interest at the time of detection.</li>
<li>This is because, in the model, the MGS reads are determined by the relative abundance of the pathogen in the sample, which is determined by the relative abundance of pathogen in the catchment; and we were interested in the prevalence (relative abundance of infected) in the catchment at time of detection, rather than absolute infection numbers.</li>
<li>Also, I assume detection occurs in a regime where stochastic effects from small infection numbers are negligible.</li>
<li>This would be very useful to know if true, since it would allow us to use estimates from a variety of WWTPs or other WW monitoring locations interchangeably.</li>
<li>Contrast with targeted detection, where the absolute concentration of the pathogen matters, and so total infected, flow rate, and effective concentration factor of the protocol, become relevant.</li>
</ul>
<p>Some caveats</p>
<ul>
<li>For these results to hold, it is important that detection occurs in the exponential quasi-equilibrium phase. This implies that the population must be large enough and detection ability is not too high.</li>
<li>I also assumed we can ignore noise from transmission and shedding; this similarly requires that there is enough absolute numbers of infections at the time of detection.</li>
</ul>
<h1 id="applications">Applications</h1>
<h2 id="example-sars2-reads-in-southern-california-wwtp-data">Example: SARS2 reads in Southern California WWTP data</h2>
<p><span class="citation" data-cites="rothman2021rnav">Rothman et al. (<a href="#ref-rothman2021rnav" role="doc-biblioref">2021</a>)</span> perform untargeted viral RNA sequencing of samples from several WWTPs in Southern California, with and without enrichment for specific viruses including SARS2.
<a href="https://docs.google.com/document/d/1XVeLyc1Peh59mILGtie8bB2eC7Mke7Byq77tiqs7Ih8/edit#heading=h.dshnx0efqzl4">Jeff previously estimated</a> that the proportion of SARS2 reads in a WWTP sample measured without enrichment is 3e-7 when the daily incidence (fraction of the population infected in a day) is 1e-3.
What can we say in general about the proportion of SARS2 reads at a given prevalence in the population?</p>
<p>First, we need to connect the current daily incidence 1e-3 to the current prevalence.
In our baseline model, the new infections per day is approximately <span class="math inline">\(b \: i(t)\)</span>, where <span class="math inline">\(b\)</span> is the rate of new infections per infected and <span class="math inline">\(i(t)\)</span> is the fraction of the population that is infected.</p>
<p>We can estimate <span class="math inline">\(b\)</span> from the rough estimates of <span class="math inline">\(R_0\)</span> and the (average) generation time compiled by Lenni last summer, where generation time is defined as</p>
<blockquote>
<p>Generation time is a modelling term describing the time duration from the onset of infectiousness in a primary case to the onset of infectiousness in a secondary case infected by the primary case. <a href="https://nccid.ca/publications/glossary-terms-infectious-disease-modelling-proposal-consistent-language/#:~:text=latent%E2%80%99%20(see%C2%A0below).-,Generation%20time,-Generation%20time%20is">source</a></p>
</blockquote>
<p>In our model, in which infectiousness is constant over the course of the disease, the average generation time is <span class="math inline">\(1/d\)</span> and <span class="math inline">\(R_0 = b/d\)</span>, so that <span class="math inline">\(b = R_0 d\)</span>.
Lenni’s point estimates were <span class="math inline">\(\hat R_0 = 3\)</span> and an average generation time of <span class="math inline">\(5.1\)</span> days (corresponding to <span class="math inline">\(\hat d = 1/5.1 \approx 0.2\)</span>, giving a point estimate for <span class="math inline">\(b\)</span> of <span class="math inline">\(\hat b = 3 / 5.1 \approx 0.59\)</span>.
(A future improvement would consider estimates specifically for this population and time period.)
At this value of <span class="math inline">\(b\)</span>, a daily incidence of <span class="math inline">\(10^{-3}\)</span> corresponds to a prevalence of <span class="math inline">\(i(t) = 10^{-3}/\hat b \approx 1.7 \cdot 10^{-3}\)</span>.</p>
<p>Therefore we have <span class="math inline">\(P = 3\cdot 10^{-7}\)</span> when <span class="math inline">\(i = 1.7 \cdot 10^{-3}\)</span>, so that <span class="math inline">\(sB = P / i = 1.8 \cdot 10^{-4} \sim 10^{-4}\)</span>.
From
<span class="math display">\[\begin{align}
  \mathcal M = \frac{M^*}{s B i(t^*) },
\end{align}\]</span>
we can compute that we would need to sequence a single sample to a depth of <span class="math inline">\(\sim 10^{9}\)</span> fragments to see 10 fragments of SARS2 when <span class="math inline">\(i = 10^{-4}\)</span> of the population is infected.</p>
<p>What about seeing an individual 40-mer in the SARS2 genome 10 times when <span class="math inline">\(i = 10^{-4}\)</span>, for 1x150bp sequencing?
In this case, roughly <span class="math inline">\(10^{-2.5}\)</span> of SARS2 reads will cover the 40-mer (<span class="math inline">\((150 - 40 + 1) / 30000\)</span>), so that we’d need to increase our sequencing depth by <span class="math inline">\(10^{2.5}\)</span>-fold relative to the case of just seeing the genome 10 times, resulting in a total sequencing depth of <span class="math inline">\(10^{11.5}\)</span> reads.</p>
<h1 id="details-and-complications">Details and complications</h1>
<p>General assumptions</p>
<ul>
<li>Local outbreak — cases come in from elsewhere (or a local release), then spread locally</li>
<li>Local wastewater treatment plant (WWTP)</li>
<li>Assume that the WWTP is a good proxy for the local population</li>
<li>Consider the exponential phase of the outbreak.</li>
</ul>
<p>Thus we need to model</p>
<ul>
<li>exponential growth (possibly noisy)</li>
<li>shedding (possibly noisy)</li>
<li>transport — possible delay, noise</li>
<li>sample collection</li>
<li>measurement via qPCR, amplicon sequencing, metagenomic sequencing</li>
</ul>
<p>Even under the simplified scenario of a local outbreak and monitoring system (closed system), there is still complication from stochasticity and complex disease time courses.
To make progress we need to look for ways to approximate the dynamics that still gives an accurate picture for the purposes of detection.
Goal of this section is to explain when and why we can approximate the dynamics of cumulative infections, active infections, and the amount of collected NA as growing deterministically and exponentially.</p>
<h2 id="complex-disease-time-courses">Complex disease time courses</h2>
<p>Qualitative model:</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://tibble.tidyverse.org/reference/tribble.html'>tribble</a></span><span class='op'>(</span></span>
<span>  <span class='op'>~</span><span class='va'>name</span>, <span class='op'>~</span><span class='va'>x</span>, <span class='op'>~</span><span class='va'>xend</span>,</span>
<span>  <span class='st'>'symptomatic'</span>, <span class='fl'>8</span>, <span class='fl'>16</span>,</span>
<span>  <span class='st'>'infectious'</span>, <span class='fl'>5</span>, <span class='fl'>12</span>,</span>
<span>  <span class='st'>'shedding'</span>, <span class='fl'>3</span>, <span class='fl'>20</span></span>
<span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>fct_inorder</span><span class='op'>)</span>,</span>
<span>    y <span class='op'>=</span> <span class='va'>name</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> <span class='va'>as.integer</span></span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, xend <span class='op'>=</span> <span class='va'>xend</span>, color <span class='op'>=</span> <span class='va'>name</span>, label <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_minimal_grid.html'>theme_minimal_hgrid</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>legend.position <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_continuous</a></span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>10</span>, <span class='fl'>20</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/expand_limits.html'>expand_limits</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fl'>0</span>, xend <span class='op'>=</span> <span class='fl'>20</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_discrete.html'>scale_y_discrete</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='va'>rev</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_brewer.html'>scale_color_brewer</a></span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>'qual'</span>, palette <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'time since exposed (days)'</span>, y <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_segment.html'>geom_segment</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>name</span>, yend <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-2-1.svg" width="1800" /></p>
</div>
<p>Quantitative model would be to have each of these represented by a continuous function of time.
Could further have a (joint) distribution of functions.</p>
<p>Can tackle this by considering the impact on the exponential growth rate and the scaling factors relating exposed to infected to recovered to shedding amounts.
A key interest is the relationship between cumulative exposures (i.e. infected) to currently infectious to shedding.</p>
<p>At large enough numbers, we should be able to approximate the effect of these complex dynamics through effective parameters linking the number of infected, number of infectious, and current rate of shedding; all will be growing asymptotically exponentially.
There are various plausible models here.
As a simple illustration, suppose that….</p>
<!--
For example, consider the following classes,

- $S$: Susceptible: Not yet exposed
- $x_{0, 0}$: Exposed, not yet shedding or infectious
- $x_{1, 0}$: Exposed, Shedding, non-infectious
- $x_{0, 1}$: Exposed, Non-shedding, infectious
- $x_{1, 1}$: Exposed, Shedding, infectious
- $R$: Recovered

We can define a transition matrix between these classes
-->
<p>A main issue left unaddressed by the SEIR model is when shedding occurs.
Also, we might not be that happy with the exponential waiting times for the disease stages.
We can address these issues by considering a more complex disease time course which explicitly includes shedding.
To remain tractable in the present framework, what is important is that it is still safe to assume that we quickly reach approximate exponential growth in the number of infected and that daily shedding is proportional to the number of infected in this asymptotic regime.</p>
<p>Questions to investigate here include</p>
<ul>
<li>what are the asymptotic growth rates and shedding proportionality constants?</li>
<li>related: how does the disease time course affect growth rate and the proportionality constant between shedding and the number of infected?</li>
<li>what is the time scale for reaching the asymptotic exponential growth regime?</li>
</ul>
<h2 id="seir-model">SEIR model</h2>
<p><strong>TODO:</strong> Consider reparameterizing to the other common form of <span class="math inline">\(dS/dt = -(\beta/N) SI\)</span>, so that we can write <span class="math inline">\(R_0 = \beta / \gamma\)</span> independent of <span class="math inline">\(N\)</span>.
I’d need to adjust the code accordingly.</p>
<p>A typical way to consider the effect of a latent (non-infectious) period is using the SEIR model.
Define the SEIR model as
<span class="math display">\[\begin{align}
  \frac{dS}{dt} &amp;= -\beta S I \\
  \frac{dE}{dt} &amp;= \beta S I - \theta E \\
  \frac{dI}{dt} &amp;= \theta E - \gamma I \\
  \frac{dR}{dt} &amp;= \gamma I
\end{align}\]</span>
where <span class="math inline">\(S\)</span> is the number of susceptible, <span class="math inline">\(E\)</span> is the number of exposed, <span class="math inline">\(I\)</span> is the number of infected, and <span class="math inline">\(R\)</span> is the number of recovered; <span class="math inline">\(\beta\)</span> is the transmission rate, <span class="math inline">\(\theta\)</span> is the incubation rate, and <span class="math inline">\(\gamma\)</span> is the recovery rate.
(This parameterization is from <span class="citation" data-cites="diekmann2012math">Diekmann, Heesterbeek, and Britton (<a href="#ref-diekmann2012math" role="doc-biblioref">2012</a>)</span>.)</p>
<p>I expect that we’ll typically be able to use a quasi-equilibrium approximation where <span class="math inline">\(E\)</span> and <span class="math inline">\(I\)</span> each grow exponentially.
Let’s work out that situation.</p>
<p>Early in the pandemic, we can take <span class="math inline">\(S \approx 1\)</span> and <span class="math inline">\(R \approx 0\)</span>, giving the approximate system
<span class="math display">\[\begin{align}
  \frac{dE}{dt} &amp;= - \theta E + B I \\
  \frac{dI}{dt} &amp;= \theta E - \gamma I,
\end{align}\]</span>
(where <span class="math inline">\(B = \beta N\)</span>)
which converges to exponential growth at rate given by
<span class="math display">\[\begin{align}
  r = \frac{-(\theta + \gamma) + \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2}.
\end{align}\]</span>
(given by the leading eigenvector <span class="math inline">\(\lambda_1\)</span> of the Jacobian matrix) (see <span class="citation" data-cites="ma2020esti">Ma (<a href="#ref-ma2020esti" role="doc-biblioref">2020</a>)</span>).
We can solve for the equilibrium ratio of exposed to infected, <span class="math inline">\(E/I\)</span>, by solving for <span class="math inline">\(d/dt(E/I) = 0\)</span>; this gives
<span class="math display">\[\begin{align}
  \frac{E}{I}
  &amp;= \frac{-(\theta - \gamma) + \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2 \theta}
\\&amp;= \frac{r + \gamma}{\theta}.
\end{align}\]</span>
(Alternatively, we could get the equilibrium ratio from the leading eigenvector of the Jacobian; see below).</p>
<p>For <span class="math inline">\(\theta (B - \gamma) \ll (\theta + \gamma)^2\)</span>, we can approximate <span class="math inline">\(r\)</span> as
<span class="math display">\[\begin{align}
  r \approx \frac{\theta (\theta - \gamma)}{\theta + \gamma},
\end{align}\]</span>
which is consistent with the results of <span class="citation" data-cites="heng2020theaa">Heng and Althaus (<a href="#ref-heng2020theaa" role="doc-biblioref">2020</a>)</span>.
<span class="citation" data-cites="heng2020theaa">Heng and Althaus (<a href="#ref-heng2020theaa" role="doc-biblioref">2020</a>)</span> focus on an approximation in which the SEIR model’s dynamics are approximately the same as the corresponding SIR model, but scaled by a factor of <span class="math inline">\(\alpha = \theta / (\theta + \gamma)\)</span>.</p>
<p>More explicitly, the eigenvalues are (<a href="https://www.wolframalpha.com/input?i=eigenvectors+%7B%7B-theta%2C+beta%7D%2C%7Btheta%2C+-gamma%7D%7D">Wolfram Alpha</a>)
<span class="math display">\[\begin{align}
  \lambda_1 &amp;= \frac{-(\theta + \gamma) + \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2} \\
  \lambda_2 &amp;= \frac{-(\theta + \gamma) - \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2}.
\end{align}\]</span>
The eigenvectors are
<span class="math display">\[\begin{align}
  v_1 &amp;= \left(\frac{-(\theta - \gamma) + \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2\theta}, 1 \right) \\
  v_2 &amp;= \left(\frac{-(\theta - \gamma) - \sqrt{(\theta - \gamma)^2 + 4 \theta B}}{2\theta}, 1 \right).
\end{align}\]</span>
I believe the (exponential) rate of convergence to the exponential quasi-equilibrium is given by the difference <span class="math inline">\(\lambda_1 - \lambda_2\)</span>,
<span class="math display">\[\begin{align}
  \lambda_1 - \lambda_2 = \sqrt{(\theta - \gamma)^2 + 4 \theta B}.
\end{align}\]</span></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># modified from http://epirecip.es/epicookbook/chapters/seir/r_desolve</span></span>
<span></span>
<span><span class='co'># Function to return derivatives of SEIR model</span></span>
<span><span class='va'>seir_ode</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>t</span>,<span class='va'>Y</span>,<span class='va'>par</span><span class='op'>)</span><span class='op'>{</span></span>
<span>  <span class='va'>S</span> <span class='op'>&lt;-</span> <span class='va'>Y</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span></span>
<span>  <span class='va'>E</span> <span class='op'>&lt;-</span> <span class='va'>Y</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span></span>
<span>  <span class='va'>I</span> <span class='op'>&lt;-</span> <span class='va'>Y</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>]</span></span>
<span>  <span class='va'>R</span> <span class='op'>&lt;-</span> <span class='va'>Y</span><span class='op'>[</span><span class='fl'>4</span><span class='op'>]</span></span>
<span>  </span>
<span>  <span class='va'>beta</span> <span class='op'>&lt;-</span> <span class='va'>par</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span></span>
<span>  <span class='va'>theta</span> <span class='op'>&lt;-</span> <span class='va'>par</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span></span>
<span>  <span class='va'>gamma</span> <span class='op'>&lt;-</span> <span class='va'>par</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>]</span></span>
<span>  </span>
<span>  <span class='va'>dYdt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/vector.html'>vector</a></span><span class='op'>(</span>length<span class='op'>=</span><span class='fl'>3</span><span class='op'>)</span></span>
<span>  <span class='va'>dYdt</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='op'>-</span><span class='va'>beta</span><span class='op'>*</span><span class='va'>I</span><span class='op'>*</span><span class='va'>S</span></span>
<span>  <span class='va'>dYdt</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>beta</span><span class='op'>*</span><span class='va'>I</span><span class='op'>*</span><span class='va'>S</span> <span class='op'>-</span> <span class='va'>theta</span><span class='op'>*</span><span class='va'>E</span></span>
<span>  <span class='va'>dYdt</span><span class='op'>[</span><span class='fl'>3</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>theta</span><span class='op'>*</span><span class='va'>E</span> <span class='op'>-</span> <span class='va'>gamma</span><span class='op'>*</span><span class='va'>I</span></span>
<span>  </span>
<span>  <span class='kw'><a href='https://rdrr.io/r/base/function.html'>return</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>dYdt</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span></span>
<span><span class='co'># Set parameter values</span></span>
<span><span class='va'>theta</span> <span class='op'>&lt;-</span> <span class='fl'>1</span><span class='op'>/</span><span class='fl'>5</span></span>
<span><span class='va'>gamma</span> <span class='op'>&lt;-</span> <span class='fl'>1</span><span class='op'>/</span><span class='fl'>10</span></span>
<span><span class='co'># Set beta from the basic reproduction number, since R0 = beta * N / gamma</span></span>
<span><span class='va'>R0</span> <span class='op'>&lt;-</span> <span class='fl'>3</span></span>
<span><span class='va'>N</span> <span class='op'>&lt;-</span> <span class='fl'>1e6</span></span>
<span><span class='va'>beta</span> <span class='op'>&lt;-</span> <span class='va'>R0</span> <span class='op'>*</span> <span class='va'>gamma</span> <span class='op'>/</span> <span class='va'>N</span></span>
<span></span>
<span><span class='co'># Set initial conditions</span></span>
<span><span class='va'>E0</span> <span class='op'>&lt;-</span> <span class='fl'>1</span></span>
<span><span class='va'>I0</span> <span class='op'>&lt;-</span> <span class='fl'>0</span></span>
<span><span class='va'>init</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>N</span> <span class='op'>-</span> <span class='va'>E0</span> <span class='op'>-</span> <span class='va'>I0</span>, <span class='va'>E0</span>, <span class='va'>I0</span><span class='op'>)</span></span>
<span><span class='va'>t</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>250</span><span class='op'>)</span></span>
<span><span class='va'>par</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>beta</span>, <span class='va'>theta</span>, <span class='va'>gamma</span><span class='op'>)</span></span>
<span><span class='co'># Solve system using lsoda</span></span>
<span><span class='va'>sol</span> <span class='op'>&lt;-</span> <span class='fu'>deSolve</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/deSolve/man/lsoda.html'>lsoda</a></span><span class='op'>(</span><span class='va'>init</span>, <span class='va'>t</span>, <span class='va'>seir_ode</span>, <span class='va'>par</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> </span>
<span>  <span class='va'>as_tibble</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://rlang.r-lib.org/reference/set_names.html'>set_names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"time"</span>, <span class='st'>"S"</span>,<span class='st'>"E"</span>,<span class='st'>"I"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>R <span class='op'>=</span> <span class='va'>N</span> <span class='op'>-</span> <span class='op'>(</span><span class='va'>S</span> <span class='op'>+</span> <span class='va'>E</span> <span class='op'>+</span> <span class='va'>I</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>r_pred</span> <span class='op'>=</span> <span class='op'>(</span><span class='op'>-</span><span class='op'>(</span><span class='va'>theta</span> <span class='op'>+</span> <span class='va'>gamma</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='op'>(</span><span class='va'>theta</span> <span class='op'>-</span> <span class='va'>gamma</span><span class='op'>)</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>+</span> <span class='fl'>4</span> <span class='op'>*</span> <span class='va'>theta</span> <span class='op'>*</span> <span class='va'>beta</span> <span class='op'>*</span> <span class='va'>N</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>2</span></span>
<span></span>
<span><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>sol</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='fu'><a href='https://tidyselect.r-lib.org/reference/everything.html'>everything</a></span><span class='op'>(</span><span class='op'>)</span>, <span class='va'>as.numeric</span><span class='op'>)</span>,</span>
<span>    I_pred <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>pmin</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='va'>r_pred</span> <span class='op'>*</span> <span class='va'>time</span><span class='op'>)</span>, <span class='va'>N</span><span class='op'>)</span>,</span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>time</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>fct_inorder</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>clrs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span>  S <span class='op'>=</span> <span class='va'>colors_oi</span><span class='op'>[[</span><span class='st'>'orange'</span><span class='op'>]</span><span class='op'>]</span>,</span>
<span>  E <span class='op'>=</span> <span class='va'>colors_oi</span><span class='op'>[[</span><span class='st'>'skyblue'</span><span class='op'>]</span><span class='op'>]</span>,</span>
<span>  I <span class='op'>=</span> <span class='va'>colors_oi</span><span class='op'>[[</span><span class='st'>'bluishgreen'</span><span class='op'>]</span><span class='op'>]</span>,</span>
<span>  R <span class='op'>=</span> <span class='va'>colors_oi</span><span class='op'>[[</span><span class='st'>'vermillion'</span><span class='op'>]</span><span class='op'>]</span>,</span>
<span>  I_pred <span class='op'>=</span> <span class='va'>colors_oi</span><span class='op'>[[</span><span class='st'>'gray'</span><span class='op'>]</span><span class='op'>]</span> </span>
<span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>x</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>time</span> <span class='op'>&lt;=</span> <span class='fl'>200</span>, <span class='va'>name</span> <span class='op'>!=</span> <span class='st'>'I_pred'</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>time</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>"Number of individuals"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-4-1.svg" width="1950" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>x</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>time</span> <span class='op'>&lt;=</span> <span class='fl'>200</span>, <span class='va'>name</span> <span class='op'>!=</span> <span class='st'>'S'</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>time</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='va'>N</span><span class='op'>)</span>, breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1e2</span>, <span class='fl'>1e4</span>, <span class='fl'>1e6</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>"Number of individuals"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-5-1.svg" width="1950" /></p>
</div>
<p>Note, I haven’t calculated the constant for the asymptotic number of infected. But we can see that we have the correct asymptotic growth rate.</p>
<p>Let’s also check our prediction for the asymptotic ratio of <span class="math inline">\(E/I\)</span> in the exponential regime,</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>ratio_pred</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='op'>-</span><span class='op'>(</span><span class='va'>theta</span> <span class='op'>-</span> <span class='va'>gamma</span><span class='op'>)</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='op'>(</span><span class='va'>theta</span> <span class='op'>-</span> <span class='va'>gamma</span><span class='op'>)</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>+</span> <span class='fl'>4</span> <span class='op'>*</span> <span class='va'>theta</span> <span class='op'>*</span> <span class='va'>beta</span> <span class='op'>*</span> <span class='va'>N</span><span class='op'>)</span><span class='op'>)</span><span class='op'>/</span><span class='op'>(</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>theta</span><span class='op'>)</span></span>
<span><span class='va'>x</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_wider.html'>pivot_wider</a></span><span class='op'>(</span><span class='va'>time</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>time</span>, <span class='va'>E</span> <span class='op'>/</span> <span class='va'>I</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='va'>ratio_pred</span>, color <span class='op'>=</span> <span class='st'>'grey'</span>, </span>
<span>             size <span class='op'>=</span> <span class='fl'>0.8</span>, linetype <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>"E / I"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-6-1.svg" width="1950" /></p>
</div>
<p>The simulation starts with 1 exposed individual; hence, the ratio of exposed to infected starts high but quickly drops to the quasi-equilibrium ratio of 1, then eventually starts to decline as the susceptible population declines and causes a drop in the rate of new infections, before appearing to asymptote at a positive value in the waning days of the epidemic.</p>
<p>Possible additions</p>
<ul class="task-list">
<li><input type="checkbox" />Expression for the peak susceptible, and total infected</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># from https://cran.r-project.org/web/packages/odin/vignettes/discrete.html</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/mrc-ide/odin'>odin</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='co'>## Core equations for transitions between compartments:</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/stats/update.html'>update</a></span><span class='op'>(</span><span class='va'>S</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>S</span> <span class='op'>-</span> <span class='va'>beta</span> <span class='op'>*</span> <span class='va'>S</span> <span class='op'>*</span> <span class='va'>I</span> <span class='op'>/</span> <span class='va'>N</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/stats/update.html'>update</a></span><span class='op'>(</span><span class='va'>E</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>S</span> <span class='op'>-</span> <span class='va'>beta</span> <span class='op'>*</span> <span class='va'>S</span> <span class='op'>*</span> <span class='va'>I</span> <span class='op'>/</span> <span class='va'>N</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/stats/update.html'>update</a></span><span class='op'>(</span><span class='va'>I</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>I</span> <span class='op'>+</span> <span class='va'>beta</span> <span class='op'>*</span> <span class='va'>S</span> <span class='op'>*</span> <span class='va'>I</span> <span class='op'>/</span> <span class='va'>N</span> <span class='op'>-</span> <span class='va'>gamma</span> <span class='op'>*</span> <span class='va'>I</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/stats/update.html'>update</a></span><span class='op'>(</span><span class='va'>R</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>R</span> <span class='op'>+</span> <span class='va'>gamma</span> <span class='op'>*</span> <span class='va'>I</span></span>
<span></span>
<span><span class='co'>## Total population size (odin will recompute this at each timestep:</span></span>
<span><span class='co'>## automatically)</span></span>
<span><span class='co'># N &lt;- S + I + R</span></span>
<span><span class='va'>N</span> <span class='op'>&lt;-</span> <span class='fl'>1</span></span>
<span></span>
<span><span class='co'>## Initial states:</span></span>
<span><span class='fu'>initial</span><span class='op'>(</span><span class='va'>S</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>S_ini</span> <span class='co'># will be user-defined</span></span>
<span><span class='fu'>initial</span><span class='op'>(</span><span class='va'>E</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>E_ini</span> <span class='co'># will be user-defined</span></span>
<span><span class='fu'>initial</span><span class='op'>(</span><span class='va'>I</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>I_ini</span> <span class='co'># will be user-defined</span></span>
<span><span class='fu'>initial</span><span class='op'>(</span><span class='va'>R</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fl'>0</span></span>
<span></span>
<span><span class='co'>## User defined parameters - default in parentheses:</span></span>
<span><span class='va'>S_ini</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>1</span> <span class='op'>-</span> <span class='fl'>1e-4</span><span class='op'>)</span></span>
<span><span class='va'>E_ini</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>0</span><span class='op'>)</span></span>
<span><span class='va'>I_ini</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>1e-4</span><span class='op'>)</span></span>
<span><span class='va'>beta</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>0.2</span><span class='op'>)</span></span>
<span><span class='va'>theta</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>0.1</span><span class='op'>)</span></span>
<span><span class='va'>gamma</span> <span class='op'>&lt;-</span> <span class='fu'>user</span><span class='op'>(</span><span class='fl'>0.1</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h2 id="random-variation-in-transmission">(Random) variation in transmission</h2>
<p>One important form of randomness is in the number of secondary infections caused by a single infection.
Discrete generation models make it particularly simple to model inter-individual variation in this number.
The Gamma-Poisson (GP) distributrion (i.e., Negative Binomial) is a convenient (and seemingly common) distributrion to use for this.
A possible justification of a Poisson mixture distribution like this is to suppose that each individual varies in how infectious they are, but conditional on their infectiousness they infect a random number of others according to the standard Poisson assumption.
Choosing a Gamma distribution for the first step is convenient because it gives us just one parameter to tune the CV in the infectiousness among individuals.
See below for more about the GP distribution and the parameterization in terms of the mean and CV of the underlying gamma distribution.</p>
<p>Very small CVs for the underlying gamma distribution amount to little variation in infectiousness.
A CV of 1 in the Gamma distribution corresponds to an exponential; higher CVs will be even fatter tailed.
In these cases, there will tend to be ‘super spreader’ infections with much higher infectiousness than the typical infection.</p>
<p>We can see the effect of stochasticity with tunable variation in infectiousness by simulating a discrete generation branching process with a GP offspring distribution.
Suppose there are <span class="math inline">\(I\)</span> infectious individuals, and each infects an iid GP number of others, with parameters <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\kappa\)</span>.
(The mean <span class="math inline">\(\mu\)</span> is also the <span class="math inline">\(R_0\)</span> of the disease.)
To simulate, we can exploit the fact that the sum of <span class="math inline">\(n\)</span> iid GP random variables with mean <span class="math inline">\(\mu\)</span> and CV of the gamma distributrion <span class="math inline">\(\kappa\)</span> is a GP random variable with mean <span class="math inline">\(n \mu\)</span> and gamma CV of <span class="math inline">\(\kappa / \sqrt{n}\)</span>.
(Note, this is somewhat a guess based on the fact that for the sum of iid Gamma rvs, the scale parameter remains the same; and the sum of Poissons is Poisson with appropriately increased mean, so we can think of summing up all the gamma’s first, then feeding into a single Poisson.)</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>next_generation</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>I0</span>, <span class='va'>mean</span>, <span class='va'>cv</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>n</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>I0</span><span class='op'>)</span></span>
<span>  <span class='va'>mean_sum</span> <span class='op'>&lt;-</span> <span class='va'>I0</span> <span class='op'>*</span> <span class='va'>mean</span></span>
<span>  <span class='va'>cv_sum</span> <span class='op'>&lt;-</span> <span class='va'>cv</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='va'>I0</span><span class='op'>)</span></span>
<span>  <span class='va'>scale_sum</span> <span class='op'>=</span> <span class='va'>cv_sum</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>mean_sum</span></span>
<span>  <span class='va'>shape_sum</span> <span class='op'>=</span> <span class='va'>cv_sum</span><span class='op'>^</span><span class='op'>(</span><span class='op'>-</span><span class='fl'>2</span><span class='op'>)</span></span>
<span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/GammaDist.html'>rgamma</a></span><span class='op'>(</span>n <span class='op'>=</span> <span class='va'>n</span>, scale <span class='op'>=</span> <span class='va'>scale_sum</span>, shape <span class='op'>=</span> <span class='va'>shape_sum</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>    <span class='op'>{</span><span class='fu'><a href='https://rdrr.io/r/stats/Poisson.html'>rpois</a></span><span class='op'>(</span>n <span class='op'>=</span> <span class='va'>n</span>, lambda <span class='op'>=</span> <span class='va'>.</span><span class='op'>)</span><span class='op'>}</span></span>
<span><span class='op'>}</span></span>
<span></span>
<span><span class='co'># Simulate many runs over many generations</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>42</span><span class='op'>)</span></span>
<span><span class='va'>num_runs</span> <span class='op'>&lt;-</span> <span class='fl'>100</span></span>
<span><span class='va'>num_gens</span> <span class='op'>&lt;-</span> <span class='fl'>20</span></span>
<span><span class='va'>mat</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fl'>0</span>, nrow <span class='op'>=</span> <span class='va'>num_gens</span>, ncol <span class='op'>=</span> <span class='va'>num_runs</span><span class='op'>)</span></span>
<span><span class='va'>mat</span><span class='op'>[</span><span class='fl'>1</span>,<span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fl'>1</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>mat</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>mat</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'>next_generation</span><span class='op'>(</span><span class='va'>mat</span><span class='op'>[</span><span class='va'>i</span><span class='op'>-</span><span class='fl'>1</span>, <span class='op'>]</span>, mean <span class='op'>=</span> <span class='fl'>2</span>, cv <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='va'>sims</span> <span class='op'>&lt;-</span> <span class='va'>mat</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> <span class='va'>as_tibble</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>time <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/row_number.html'>row_number</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>-</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>time</span>, names_to <span class='op'>=</span> <span class='st'>'.id'</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>         <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>.id</span>, <span class='va'>str_sub</span>, <span class='fl'>2</span><span class='op'>)</span>,</span>
<span>         <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>.id</span>, <span class='va'>as.numeric</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>mat</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sims</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>.id</span> <span class='op'>&lt;=</span> <span class='fl'>20</span>, <span class='va'>time</span> <span class='op'>&lt;=</span> <span class='fl'>15</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>time</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>as.factor</a></span><span class='op'>(</span><span class='va'>.id</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>legend.position <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'Time (generations)'</span>, y <span class='op'>=</span> <span class='st'>'Infections'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>1e2</span>, <span class='fl'>1e3</span>, <span class='fl'>1e4</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='fl'>2</span><span class='op'>)</span> <span class='op'>*</span> <span class='va'>time</span><span class='op'>)</span><span class='op'>)</span>, color <span class='op'>=</span> <span class='st'>'black'</span>, linetype <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-10-1.svg" width="1950" /></p>
</div>
<p>In the simulations above (R0 of 2, CV of 0.3), we see that for at or below ~10 infections, stochasticity is important; but for much above, we have approximately deterministic exponential growth.</p>
<p>In a superspreader scenario (R0 of 2, CV of 2),</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Simulate many runs over many generations</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>42</span><span class='op'>)</span></span>
<span><span class='va'>num_runs</span> <span class='op'>&lt;-</span> <span class='fl'>100</span></span>
<span><span class='va'>num_gens</span> <span class='op'>&lt;-</span> <span class='fl'>20</span></span>
<span><span class='va'>mat</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fl'>0</span>, nrow <span class='op'>=</span> <span class='va'>num_gens</span>, ncol <span class='op'>=</span> <span class='va'>num_runs</span><span class='op'>)</span></span>
<span><span class='va'>mat</span><span class='op'>[</span><span class='fl'>1</span>,<span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fl'>1</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>mat</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>mat</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span> <span class='op'>&lt;-</span> <span class='fu'>next_generation</span><span class='op'>(</span><span class='va'>mat</span><span class='op'>[</span><span class='va'>i</span><span class='op'>-</span><span class='fl'>1</span>, <span class='op'>]</span>, mean <span class='op'>=</span> <span class='fl'>2</span>, cv <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='va'>sims</span> <span class='op'>&lt;-</span> <span class='va'>mat</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> <span class='va'>as_tibble</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>time <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/row_number.html'>row_number</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>-</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>time</span>, names_to <span class='op'>=</span> <span class='st'>'.id'</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>         <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>.id</span>, <span class='va'>str_sub</span>, <span class='fl'>2</span><span class='op'>)</span>,</span>
<span>         <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>.id</span>, <span class='va'>as.numeric</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>mat</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sims</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>.id</span> <span class='op'>&lt;=</span> <span class='fl'>20</span>, <span class='va'>time</span> <span class='op'>&lt;=</span> <span class='fl'>15</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>time</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>as.factor</a></span><span class='op'>(</span><span class='va'>.id</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span>legend.position <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'Time (generations)'</span>, y <span class='op'>=</span> <span class='st'>'Infections'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>1e2</span>, <span class='fl'>1e3</span>, <span class='fl'>1e4</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='fl'>2</span><span class='op'>)</span> <span class='op'>*</span> <span class='va'>time</span><span class='op'>)</span><span class='op'>)</span>, color <span class='op'>=</span> <span class='st'>'black'</span>, linetype <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-12-1.svg" width="1950" /></p>
</div>
<p>we see that stochasticity remains relevant for larger sizes of 100 to 300 infections.
We can also more strongly see another expected effect of randomness, which is that conditional on survival, the outbreak grows faster than the asymptotic rate initially.
Notably, however, in both cases the asymptotic exponential growth rate is the same (<span class="math inline">\(\log R_0\)</span>).</p>
<p>Remaining questions/todos include</p>
<ul>
<li>Establishment size above which an outbreak is approximately guaranteed and can ignore stochastic effects; what is this size for real epidemics?</li>
<li>I’m used to models where the point at which success is assured and which can ignore stochastic effects are the same; is that also true with very skewed offspring/infection distributions?</li>
<li>Expression for the CV of <span class="math inline">\(I(t)\)</span></li>
</ul>
<h3 id="relevant-references">Relevant references</h3>
<p><span class="citation" data-cites="morantovar2022stoc">Morán-Tovar et al. (<a href="#ref-morantovar2022stoc" role="doc-biblioref">2022</a>)</span> section 2.2 discusses stochasticity in transmission, but without variation in infectiousness among individuals.
The model there is equivalent to what Charlie, Dan, and Jake have been using.
<span class="citation" data-cites="morantovar2022stoc">Morán-Tovar et al. (<a href="#ref-morantovar2022stoc" role="doc-biblioref">2022</a>)</span> section 2.3 uses a contact network model to consider heterogeneity in individuals.
Super-spreaders emerge from highly-connected nodes.</p>
<h2 id="origination-of-the-outbreak">Origination of the outbreak</h2>
<p>Suppose new infections coming in at rate <span class="math inline">\(M\)</span>.
Can understand the point at which we have a deterministic local outbreak and can ignore new imports, based on what we know about this from evolutionary theory.</p>
<h2 id="shedding">Shedding</h2>
<p>Simple model used in the modeling literature links cases with shedding by supposing that there is a fixed distribution of shedding as a function of time post symptom onset.
The idea that everyone sheds the same can be misleading; and is particularly sketchy in cases where there is substantial heterogeneity in how people respond to the disease.
Still, it is a fair starting point and useful for illustration.</p>
<p>In our case, we’re interested in shedding that occurs even without clinical presentation, so I’ll consider that there is a shedding distribution <span class="math inline">\(s(t)\)</span> indicating the amount of shedding from a person <span class="math inline">\(t\)</span> days after infection (exposure).
If the number of exposures is <span class="math inline">\(E(t)\)</span>, then the total shedding <span class="math inline">\(S(t)\)</span> at time <span class="math inline">\(t\)</span> is given by the convolution of <span class="math inline">\(E\)</span> with <span class="math inline">\(s\)</span>,
<span class="math display">\[\begin{align}
  S(t) = \int_{-\infty}^{\infty} E(u) \; s(t - u) \; du.
\end{align}\]</span>
Suppose the number of exposed grows exponentially at rate <span class="math inline">\(r\)</span>, <span class="math inline">\(E(t) = E_0 e^{rt}\)</span>.
We know that <span class="math inline">\(s(t) = 0\)</span> for <span class="math inline">\(t \le 0\)</span>, and we further expect <span class="math inline">\(s\)</span> to initially increase, peak at some time <span class="math inline">\(t^*\)</span>, and then decay to 0.
Let’s take it to be Gamma distributed with a mean of 10 days and a CV of 0.5,</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>shedding</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span>mean <span class='op'>=</span> <span class='fl'>10</span>, cv <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    sd <span class='op'>=</span> <span class='va'>cv</span> <span class='op'>*</span> <span class='va'>mean</span>,</span>
<span>    var <span class='op'>=</span> <span class='va'>sd</span><span class='op'>^</span><span class='fl'>2</span>,</span>
<span>    scale <span class='op'>=</span> <span class='va'>var</span> <span class='op'>/</span> <span class='va'>mean</span>,</span>
<span>    shape <span class='op'>=</span> <span class='va'>mean</span> <span class='op'>/</span> <span class='va'>scale</span></span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/expand.html'>crossing</a></span><span class='op'>(</span>t <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>30</span>, by <span class='op'>=</span> <span class='fl'>0.1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    density <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/GammaDist.html'>dgamma</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>t</span>, shape <span class='op'>=</span> <span class='va'>shape</span>, scale <span class='op'>=</span> <span class='va'>scale</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span></span>
<span><span class='va'>shedding</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>density</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'Time (days) after exposure'</span>, y <span class='op'>=</span> <span class='st'>'Density of shedding'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-13-1.svg" width="1950" /></p>
</div>
<p>Here, total area under the curve is 1, so we can think of the units as in terms of the total shedding over a single infection.</p>
<p>Now let’s view the total shedding rate over time, starting from a single exposed and under a growth rate of <span class="math inline">\(r = 0.1\)</span>,</p>
<p><span class="math display">\[\begin{align}
  S(t) = \int_{0}^{t} e^{r u} \; s(t - u) \; du.
\end{align}\]</span></p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>total_shedding</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>r</span>, <span class='va'>gamma_mean</span>, <span class='va'>gamma_cv</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>scale</span> <span class='op'>=</span> <span class='va'>gamma_cv</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>gamma_mean</span></span>
<span>  <span class='va'>shape</span> <span class='op'>=</span> <span class='va'>gamma_cv</span><span class='op'>^</span><span class='op'>(</span><span class='op'>-</span><span class='fl'>2</span><span class='op'>)</span></span>
<span>  <span class='va'>f</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>u</span><span class='op'>)</span> <span class='op'>{</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='va'>r</span> <span class='op'>*</span> <span class='va'>u</span><span class='op'>)</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/stats/GammaDist.html'>dgamma</a></span><span class='op'>(</span><span class='va'>t</span> <span class='op'>-</span> <span class='va'>u</span>, shape <span class='op'>=</span> <span class='va'>shape</span>, scale <span class='op'>=</span> <span class='va'>scale</span><span class='op'>)</span> <span class='op'>}</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/integrate.html'>integrate</a></span><span class='op'>(</span><span class='va'>f</span>, lower <span class='op'>=</span> <span class='fl'>0</span>, upper <span class='op'>=</span> <span class='va'>t</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='co'># total_shedding(10, 0.1, 10, 0.5)$value</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>r</span> <span class='op'>&lt;-</span> <span class='fl'>0.1</span></span>
<span><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span>t <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    Exposed <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='va'>r</span> <span class='op'>*</span> <span class='va'>t</span><span class='op'>)</span>,</span>
<span>    Shed <span class='op'>=</span> <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map</a></span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>total_shedding</span>, r <span class='op'>=</span> <span class='va'>r</span>, gamma_mean <span class='op'>=</span> <span class='fl'>10</span>, gamma_cv <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span>,</span>
<span>    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>Shed</span>, <span class='va'>map_dbl</span>, <span class='st'>'value'</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>x</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>t</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1e-2</span>, <span class='cn'>NA</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-16-1.svg" width="1950" /></p>
</div>
<p>After an initial lag, the shedding rate too grows exponentially, as expected.</p>
<p>We can attempt a rough calculation of the ratio ratio <span class="math inline">\(S/E\)</span>:</p>
<ul>
<li>The total shedding amount over an infection is 1</li>
<li>The bulk of the shedding of the infected is roughly between days 5 and 15;</li>
<li>So <span class="math inline">\(S(t)\)</span> is roughly the sum of contributions from those infected over 5-15 days in the past, and each of them are contributing 1/10. (The mean <span class="math inline">\(\pm\)</span> the standard deviation)</li>
<li>5-15 days in the past is (in this case) a period of <span class="math inline">\(-1/2r\)</span> to <span class="math inline">\(-3/2r\)</span>; the integral of <span class="math inline">\(e^{rt}\)</span> over that period is <span class="math inline">\(1/(10 r) e^{rt} (e^{-1/2} - e^{-3/2})\)</span></li>
<li>This suggests a ratio of <span class="math inline">\((e^{-1/2} - e^{-3/2})/(10 r)\)</span> or 0.38.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>ratio_guess</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>/</span><span class='fl'>2</span><span class='op'>)</span> <span class='op'>-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>3</span><span class='op'>/</span><span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> <span class='op'>(</span><span class='fl'>10</span> <span class='op'>*</span> <span class='va'>r</span><span class='op'>)</span></span>
<span><span class='va'>x</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>Guess <span class='op'>=</span> <span class='va'>Exposed</span> <span class='op'>*</span> <span class='va'>ratio_guess</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>t</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>value</span>, color <span class='op'>=</span> <span class='va'>name</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1e-2</span>, <span class='cn'>NA</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-17-1.svg" width="1950" /></p>
</div>
<p>We also expect there to be substantial variance in shedding among and within individuals; what effect does this have on total shedding over the course of a day?
The distinction between variance within versus between individuals is not fully defined outside the context of a specific model, but roughly I mean the following.
By variance among individuals, I mean the fact that individuals may respond differently to the pathogen in a manner that causes some to shed much more than others, or to have a significantly different time course.
By variance within individuals, I mean that how much any one person sheds on a given day (or exactly when within the day) might show additional variation beyond what can be explained from these broader differences.</p>
<p>Roughly speaking, we expect the variance in shedding to be negligible when the number of individuals who contribute to a sample is large enough.
How large is enough?</p>
<p>It is reasonable to suppose that each infected person sheds independentally, and therefore the CV in the total amount shed on a given day scales as <span class="math inline">\(1/\sqrt{n}\)</span>, where <span class="math inline">\(n\)</span> is the number of infected shedders.
However, each infected shedder does not contribute equally to shedding — any ‘super shedders’ or people who are in the peak shedding period will contribute (on average) much more than others.
For a given assumption about the shedding timecourse and how it varies among infected individuals, there will be some effective number of shedders, <span class="math inline">\(n_e\)</span>, which will be less than the total number of shedders <span class="math inline">\(n\)</span>, that captures how the <span class="math inline">\(CV\)</span> decreases.
At quasi-equilibrium, there will be a fixed distribution of shedders and <span class="math inline">\(n_e \propto n\)</span>.</p>
<p>What we ultimately care about is the shedding contributing to an individual WWTP sample.
A 24h composite sample will integrate over shedding from more than the past 24h, which will tend to buffer the day to day variation and thus increase the corresponding <span class="math inline">\(n_e\)</span> to some degree.
Our current guess is that most shedding will be arrive at the sampling site within 24h, in which case this effect will be small.</p>
<p>We can get a sense of the effective number of shedders contributing to a WWTP sample from studies that have attempted to link SARS2 or influenza WWTP levels to local prevalence, divided by the mean daily shedding rate observed in studies that assess fecal shedding in individuals.</p>
<h2 id="transport">Transport</h2>
<p>It takes time for the waste to travel to the WWTP, (I think) typically on the order of hours to days.
This time depends on the location and its distance from the WWTP, along with other factors such as the typical flow rates in the various pipes and effects on the flow such as usage and rain.
At a most basic level, the effect of transport is to create a lag between shedding and MGS signal and therefore to delay when an outbreak is detected by a corresponding amount.
But perhaps more important is what happens to the waste during this time.
I address both in turn.</p>
<h3 id="effect-of-transport-time">Effect of transport time</h3>
<p>It takes time for the waste to travel to the WWTP.
How does this affect the pathogen relative abundance at the point of sampling?</p>
<p>Complications include
* Shedding will be taking place all over the city, at different times, randomly.
* The signal from a given shedding event may spread out over days.
* But in the quasi-exponential regime, I guess that we can capture these complications by an effective recovery parameter.
* Rain and/or other large perturbations to the WW system. Can we capture this with an effective noise (CV) parameter?</p>
<p>As a way to get intuition, suppose that the amount being shed into the WW system at time <span class="math inline">\(t\)</span> is given by <span class="math inline">\(S(t)\)</span> and that the fraction of NA collected in the sample a time <span class="math inline">\(t\)</span> after it is shed is given by <span class="math inline">\(w(t)\)</span>.
The amount of NA collected at time <span class="math inline">\(t\)</span>, <span class="math inline">\(W(t)\)</span>, is given by the convolution of <span class="math inline">\(S\)</span> and <span class="math inline">\(w\)</span>,
<span class="math display">\[\begin{align}
  W(t) = \int_{-\infty}^{\infty} S(u) \; w(t - u) \; du.
\end{align}\]</span>
Our main interest is in <span class="math inline">\(W(t)\)</span> when the shedding amount grows exponentially at rate <span class="math inline">\(r\)</span>, <span class="math inline">\(S(t) = S_0 e^{rt}\)</span>.
We know that <span class="math inline">\(w(t) = 0\)</span> for <span class="math inline">\(t \le 0\)</span>, and we further expect <span class="math inline">\(w\)</span> to initially increase, peak at some time <span class="math inline">\(t^*\)</span>, and then decay to 0.
For WWTPs, I expect the peak to be on the order of a day or less (<span class="math inline">\(t^* \lesssim 1\)</span>), and the timescale of the decay to be on the order of days, e.g. the half life around 0.5 to 3 days, but I haven’t looked into this carefully and this intuition is mostly based on the Helsinki study.</p>
<p>Intuition for shape of <span class="math inline">\(w(t)\)</span>:</p>
<ul>
<li>Could try modeling as flow with turbulent diffusion -&gt; model concentration with Brownian motion with a constant ‘drift’ term representing the average flow rate. Should be fairly easy to simulate and understand the timescale and tail shape with formal solution or heuristics for this approximation.</li>
<li>Complication is that the transit length will vary by shedding location, and that the flow rate will vary by shedding location as well as along the transport path. The diffusion rate would also vary.</li>
<li>May be a long tail of arrival due to low-turnover reservoirs in the system where things might hang out; however, I don’t expect these to be very important contributors in an exponential growth scenario.</li>
<li>Perturbances such as a rain storm may also be important.</li>
</ul>
<p>It still seems useful to consider the simple model where there is a given transit length <span class="math inline">\(l\)</span>.
Brownian motion at rate <span class="math inline">\(\sigma\)</span> with drift at rate <span class="math inline">\(v\)</span>, where the transit length is <span class="math inline">\(l\)</span>, gives<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> that the concentration at time <span class="math inline">\(t\)</span> is normally distributed with mean <span class="math inline">\(vt\)</span> and standard deviation <span class="math inline">\(\sigma \sqrt{t}\)</span>.
The concentration at the sampling point (location <span class="math inline">\(l\)</span> from the shedding location) is
<span class="math display">\[\begin{align}
  w_l(t) = \frac{1}{\sqrt{2 \pi \sigma^2 t}} \exp \left[\frac{(l - v t)^2}{2\sigma^2 t} \right].
\end{align}\]</span>
The concentration peak occurs at <span class="math inline">\(t^* = l / v\)</span>.</p>
<p>TODO: Determine the decay behavior</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>v</span> <span class='op'>=</span> <span class='fl'>1</span></span>
<span><span class='va'>l</span> <span class='op'>=</span> <span class='fl'>1</span></span>
<span><span class='va'>sigma</span> <span class='op'>=</span> <span class='fl'>0.2</span></span>
<span><span class='va'>t_star</span> <span class='op'>=</span> <span class='va'>l</span><span class='op'>/</span><span class='va'>v</span></span>
<span><span class='va'>sigma_star</span> <span class='op'>=</span> <span class='va'>sigma</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='va'>t_star</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span></span>
<span>  t <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>2</span>, by <span class='op'>=</span> <span class='fl'>0.01</span><span class='op'>)</span></span>
<span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    w <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>dnorm</a></span><span class='op'>(</span><span class='va'>l</span>, mean <span class='op'>=</span> <span class='va'>v</span> <span class='op'>*</span> <span class='va'>t</span>, sd <span class='op'>=</span> <span class='va'>sigma</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='va'>t</span><span class='op'>)</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>t</span>, <span class='va'>w</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_vline</a></span><span class='op'>(</span>xintercept <span class='op'>=</span> <span class='va'>t_star</span>, color <span class='op'>=</span><span class='st'>'grey'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotate.html'>annotate</a></span><span class='op'>(</span>geom <span class='op'>=</span> <span class='st'>'segment'</span>, color <span class='op'>=</span> <span class='st'>'grey'</span>,</span>
<span>           x <span class='op'>=</span> <span class='va'>t_star</span> <span class='op'>-</span> <span class='va'>sigma_star</span>,</span>
<span>           xend <span class='op'>=</span> <span class='va'>t_star</span> <span class='op'>+</span> <span class='va'>sigma_star</span>,</span>
<span>           y <span class='op'>=</span> <span class='fl'>1</span>, yend <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-18-1.svg" width="1950" /></p>
</div>
<p>Note that the distribution is skewed, with more weight on the right side of the peak.</p>
<p>Let’s now combine this with the shedding function <span class="math inline">\(S(t)\)</span> to consider the total amount of NA collected at <span class="math inline">\(t\)</span>,</p>
<p><span class="math display">\[\begin{align}
  W(t) &amp;= \int_{-\infty}^{\infty} S(u) \; w(t - u) \; du \\
       &amp;= \int_{-\infty}^{\infty} S_0 \frac{1}{\sqrt{2 \pi \sigma^2 (t-u)}}
          \exp \left[ru + \frac{(l - v (t-u))^2}{2\sigma^2 (t-u)}\right] \; du
\end{align}\]</span></p>
<p>My guess is we can do an ok job of understanding the behavior with a heuristic approximation, where we consider the lag time <span class="math inline">\(t^* = l/v\)</span> and the spread at this time approximated by the standard deviation <span class="math inline">\(\sigma \sqrt{t^*}\)</span>.</p>
<p>My current very rough guess is that typically, most of the shed material passes through the WWTP influent within 24h, and that the vast majority (i.e., <span class="math inline">\(\ge 90\%\)</span>) passes through the WWTP within 4 days.
This guess is based on</p>
<ul>
<li>I recall the Helsinki poliovirus deposition experiment saw a peak within 1-2 days</li>
<li>Studies that attempt to link WWTP qPCR levels of SARS2 or flu to community infections so far seem to have largely ignored the WWTP transit time, focusing instead on the delay caused by the timecourse of shedding
<ul>
<li><a href="https://www.sciencedirect.com/science/article/pii/S0048969721051962" class="uri">https://www.sciencedirect.com/science/article/pii/S0048969721051962</a></li>
<li><a href="https://ehp.niehs.nih.gov/doi/full/10.1289/EHP10050" class="uri">https://ehp.niehs.nih.gov/doi/full/10.1289/EHP10050</a></li>
<li><a href="https://www.medrxiv.org/content/10.1101/2023.01.23.23284894v1" class="uri">https://www.medrxiv.org/content/10.1101/2023.01.23.23284894v1</a></li>
</ul></li>
<li><span class="citation" data-cites="soller2022mode">Soller et al. (<a href="#ref-soller2022mode" role="doc-biblioref">2022</a>)</span> fit a lognormal distribution to the times reported by <span class="citation" data-cites="kapo2017esti">Kapo et al. (<a href="#ref-kapo2017esti" role="doc-biblioref">2017</a>)</span>, finding a lognormal distribution in hours with parameters (1.2, 0.85), which are the mean and variance of the underlying normal distribution and have units of natural-log hours. This corresponds to a geometric mean of 3.3h and a geometric sd of 2.3h.</li>
<li>An on-the-fly calculation done by Lenni using <a href="https://www.aboutcivil.org/various-flow-rates-definitions.html">this site</a> to estimate it would take ~4h for WW to flow from Kendall Square to Deer Island WWTP</li>
</ul>
<p><span class="citation" data-cites="soller2022mode">Soller et al. (<a href="#ref-soller2022mode" role="doc-biblioref">2022</a>)</span> notes that, for samples of settled solids, the time the pathogen spends in the WWTP before sample collection may be larger than the time spent traveling to the WWTP.</p>
<p>2023-03-31: I’ve updated towards transit times being significantly longer than what is suggested above for our local WWTP, around 12h for Cambridge and up to ~3 days for the farthest locations (Framingham).</p>
<h3 id="changes-in-waste-composition-during-transport">Changes in waste composition during transport</h3>
<p>During transport, there will be live biological and chemical activity, which can affect the composition of the sample collected and in particular the relative abundance of the pathogen.
The relative abundance of the pathogen NA (the proportion of pathogen NA among total NA) will change due to</p>
<ol type="1">
<li>Growth of the pathogen is possible but does not seem relevant for most pathogens, which I expect to only replicate in human cells, but could be true for some pathogens that can replicate in the GI tract or the environment</li>
<li>Degredation or decay of the pathogen, either due to chemical reactions, physical forces, or biological factors (viruses and their NA may be destroyed or consumed by microbes; microbes may be killed by viruses or other microbes).</li>
<li>Growth and other biological activity of the background and sewage microbiome. This will tend to decrease the relative abundance of the pathogen.</li>
<li>Degredation or decay of the background. This too could be substantial and would serve to increase the relative abundances of the pathogen.</li>
</ol>
<p>Growth and metabolism of members of the microbiome that are more adapted to the sewer environment may may substantially lower the relative abundance of the pathogen (and thus its MGS reads) via purely compositional effects (increase in background abudance lowers pathogen relative abundance) and via increasing the decay of the pathogen through destruction/consumption.</p>
<p>There may also be a resident sewer microbiome, which could be contributing an additional amount of non-pathogen NA and adding to the decay of the pathogen.</p>
<p>Note, the pathogen and background will be changing physical form and it isn’t obvious how that will affect the resulting MGS data. E.g. a virus can be killed or inactivated but still yield NA.</p>
<p>The period in the primary clarifier may also be important for solids samples, depending on how long that time is and how more/less hospitable that setting is.</p>
<p>We can write out the relevant math. The contribution of the native sewage microbiome can be modeled by a fixed additional contribution, which will have different MGS bias. The differences in growth and decay can be modeled to first approximation with organism-dependent bias factors that scale with transport time, though this approximation might have important limitations.</p>
<p>We can try lit review to consider how we can expect various organisms to behave during transport.
I expect that the math will tell us that it is the pathogen’s growth/decay, and the most abundant (weighted in terms of <span class="math inline">\(sB\)</span>) of the background organisms that most matter here.</p>
<h2 id="sample-collection">Sample collection</h2>
<p>Consider implications of different sample types, particularly influent versus settled solids.</p>
<h3 id="partitioning-into-solid-and-liquid-fractions">Partitioning into solid and liquid fractions</h3>
<p>How the virus and the background partition into the solid versus liquid fraction will influence the relative bias parameter <span class="math inline">\(B\)</span>.
Further, if the partitioning is non-linear, then it will effectively make <span class="math inline">\(B\)</span> no longer a constant.</p>
<p><span class="citation" data-cites="kim2022sars">Kim et al. (<a href="#ref-kim2022sars" role="doc-biblioref">2022</a>)</span> fit a version of the Freundlich isotherm model,</p>
<p><span class="math display">\[\begin{align}
  C_s = K_f C_l^{1/n},
\end{align}\]</span></p>
<p>where</p>
<ul>
<li><span class="math inline">\(C_s\)</span> is the concentration in the solid fraction (copies per g)</li>
<li><span class="math inline">\(C_l\)</span> is the concentration in the liquid fraction (copies per mL)</li>
<li><span class="math inline">\(K_f\)</span> is a coefficient (ml per g)</li>
<li><span class="math inline">\(n\)</span> is a coefficient (unitless) that determines the exponent <span class="math inline">\(1/n\)</span> and hence linearity of the relationship between <span class="math inline">\(C_s\)</span> and <span class="math inline">\(C_l\)</span></li>
</ul>
<p>They fit this model using measured concentration in influent for <span class="math inline">\(C_l\)</span> and the measured concentration in settled solids for <span class="math inline">\(C_s\)</span>, using linear regression of <span class="math inline">\(\log C_s\)</span> against <span class="math inline">\(\log C_l\)</span>.
They do so separately for 5 treatment plants, finding results consistent with <span class="math inline">\(n \in [2,3]\)</span> and <span class="math inline">\(K_f \in [10^3, 10^5]\)</span> mL g<span class="math inline">\(^{-1}\)</span>.
Notably, the fact that <span class="math inline">\(\hat n \ge 2\)</span> suggests that the relationship between <span class="math inline">\(C_s\)</span> and <span class="math inline">\(C_l\)</span> is not linear; instead, <span class="math inline">\(C_s\)</span> increases more slowly than <span class="math inline">\(C_l\)</span>.
Note: When I examined the data by eye, I observed a poor fit which suggests we shouldn’t take these results too literally.</p>
<p>It is hard to say what this implies about MGS measurements without also knowing how the concentration of the background behaves.
This study also performs measurments of PMMoV; it and related Tobamaviruses are important members of the background for RNA-based protocols.
However, non-enveloped viruses as well as bacteria are important background members which we might expect to behave quite differently.</p>
<p>Future studies that use metagenomics of paired influent and solids samples can consider how the <span class="math inline">\(K_f\)</span> and <span class="math inline">\(n\)</span> parameters vary across different types of organisms.</p>
<h2 id="detection-methods">Detection methods</h2>
<h3 id="metagenomic-sequencing">Metagenomic sequencing</h3>
<p>A few features of metagenomic sequencing to be aware of</p>
<ol type="1">
<li>To first approximation under standard MGS protocols, the number of reads of the pathogen depends on how deeply we sequence overall and on the relative abundance of the pathogen NA in the sample, <em>not</em> the absolute amount or concentration of pathogen NA.</li>
<li>MGS measurements are taxonomically biased; there can also be bias within a genome</li>
<li>MGS measurements are noisy and come as discrete counts.</li>
</ol>
<p>The noise (or variability) associated with the random counting process associated with sequencing is typically modeled as Poisson or multinomial.
Which isn’t very important, as the multinomial is what you get if you suppose that there are a bunch of different species or other taxonomic features, all of which are independent Poisson counts, and you’ve conditioned on the sum (total count).</p>
<p>For the purposes of this document, I only consider the reads mapping to a single pathogen or subsequence and so ignore the multivariate nature of the MGS measurement.</p>
<p>The Poisson model only captures a part of the variance associated with the MGS measurement.
It is helpful (and more accurate) to think of the read count of the pathogen as coming from a two step process.
Starting from a WW sample, we process the sample to generate a sequencing library.
Then, we sequence the sample, giving a Poisson-distributed number of reads.
Each step entails some bias; the first step also adds noise beyond the Poisson noise.
If we model the bias in the Poisson step as a binomial filter, the result remains Poisson and is equivalent to multiplying the mean (<span class="math inline">\(\lambda\)</span>) parameter of the Poisson distribution by a bias factor.</p>
<p>A convenient and perhaps natural choice (explained below) is to model the first step by a Gamma distribution, and the second step as a Poisson distribution with mean (<span class="math inline">\(\lambda\)</span>) parameter equal to the Gamma rv.
In this context, it is natural to parameterize the Gamma distribution by its mean
mean <span class="math inline">\(\mu\)</span> and coefficient of variation <span class="math inline">\(\kappa\)</span>.
(The more common shape and scale parameters are shape of <span class="math inline">\(1/\kappa^2\)</span> and scale of <span class="math inline">\(\mu \kappa^2\)</span>.)
This parameterization allows us to separately consider bias (or effect on the expected count) through <span class="math inline">\(\mu\)</span> and the added noise from sample processing through <span class="math inline">\(\kappa\)</span>.
<!--
The coefficient of variation is a more helpful way to quantify variation than the variance or standard deviation for any attempts to quantify multiplicative (e.g. exponential growth) dynamics.
-->
We can plot the density function of the Gamma distribution for a fixed mean and different CVs ranging from above and below 1,</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>mu</span> <span class='op'>&lt;-</span> <span class='fl'>1</span></span>
<span><span class='va'>gamma_dists</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span>mean <span class='op'>=</span> <span class='va'>mu</span>, cv <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.1</span>, <span class='fl'>0.3</span>, <span class='fl'>1</span>, <span class='fl'>3</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    <span class='co'># convert from mean and cv to shape and scale params</span></span>
<span>    <span class='co'># sd = cv * mean,</span></span>
<span>    <span class='co'># var = sd^2,</span></span>
<span>    <span class='co'># scale = var / mean,</span></span>
<span>    <span class='co'># shape = mean / scale</span></span>
<span>    scale <span class='op'>=</span> <span class='va'>cv</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>mean</span>,</span>
<span>    shape <span class='op'>=</span> <span class='fl'>1</span> <span class='op'>/</span> <span class='va'>cv</span><span class='op'>^</span><span class='fl'>2</span></span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/expand.html'>crossing</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>3</span> <span class='op'>*</span> <span class='va'>mu</span>, by <span class='op'>=</span> <span class='fl'>0.02</span> <span class='op'>*</span> <span class='va'>mu</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    density <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/GammaDist.html'>dgamma</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>x</span>, shape <span class='op'>=</span> <span class='va'>shape</span>, scale <span class='op'>=</span> <span class='va'>scale</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span></span>
<span></span>
<span><span class='va'>clrs</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span>name<span class='op'>=</span><span class='st'>"Greens"</span>, n<span class='op'>=</span><span class='fl'>5</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span></span>
<span><span class='va'>gamma_dists</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>cv</span>, <span class='va'>as.factor</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>x</span> <span class='op'>/</span> <span class='va'>mu</span>, <span class='va'>density</span>, color <span class='op'>=</span> <span class='va'>cv</span>, linetype <span class='op'>=</span> <span class='va'>cv</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span></span>
<span>    axis.text.y <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_blank</a></span><span class='op'>(</span><span class='op'>)</span>,</span>
<span>    axis.ticks.y <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/element.html'>element_blank</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span>  <span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='co'># scale_color_brewer(type = 'seq', palette = 5,</span></span>
<span>  <span class='co'>#                    values = ~scales::rescale(.x, to = c(0, 0.5))) +</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'x / E[x]'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>0.8</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-19-1.svg" width="1440" /></p>
</div>
<p>This plot was generated with a mean of 1; however, the shape of the distribution does not depend on the mean, only the CV (which determines the ‘shape’ parameter under the traditional parameterization).
A CV of 1 corresponds to an exponential distribution, which has it’s mode at 0.
Higher CVs corresponds to distributions that are even more sharply peaked at 0.
As the CV decreases, the distribution becomes like a normal distribution with a shrinking standard deviation.
Changing the mean without changing the CV rescales the entire distribution, keeping the same overall shape.</p>
<p>The CV of 1 looks quite different in form, and a CV of 3 is already quite extreme.
For CVs of 3 or more, we begin to have large consequences for probability of detection (see below).
It will be helpful to find upper bounds on the CV to ensure we can ignore complications that arise when the CV is similar or larger than 1.</p>
<p>The read count <span class="math inline">\(M\)</span> then has a Gamma-Poisson (GP) distribution, also known as a Negative Binomial distribution.
The mean <span class="math inline">\(E[M]\)</span> under the GP model is the same as that of the underlying Gamma distribution, <span class="math inline">\(\mu\)</span>.
The CV of <span class="math inline">\(M\)</span> is
<span class="math display">\[\begin{align}
  \text{CV}[M] &amp;= \sqrt{\frac{1}{\mu} + \kappa^2}
  \\&amp;=
  \begin{cases}
    \frac{1}{\sqrt{\mu}} &amp; \text{for $\mu \ll \kappa$} \\
    \kappa &amp; \text{for $\mu \gg \kappa$}.
  \end{cases}
\end{align}\]</span>
The CV is approximately the larger of <span class="math inline">\(\kappa\)</span>, the CV of the Gamma distribution, and <span class="math inline">\(1/\sqrt{mu}\)</span>, which is the CV of a Poisson variable with mean <span class="math inline">\(\mu\)</span>.</p>
<p>Next</p>
<ul>
<li>Add in bias and sequencing depth</li>
<li>Explain how bias in sample prep and after can both be captured by the <span class="math inline">\(B\)</span> parameter</li>
<li>Perhaps down below, consider the effect of noise on detection i.e. the probability of seeing a certain number of reads</li>
<li>Covering a target</li>
<li>In future version, perhaps put GP definition and features in an Appendix</li>
</ul>
<p>Probability of seeing zero reads as a function of the mean or CV parameter.
Can look at using the negative binomial distribution in R; set the ‘mu’ parameter equal to the mean of the gamma rv, and the ‘size’ parameter equal to the shape (or <span class="math inline">\(1/\kappa^2\)</span>) of the gamma rv.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>nb_prob</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://tidyr.tidyverse.org/reference/expand.html'>crossing</a></span><span class='op'>(</span>mean <span class='op'>=</span> <span class='fl'>10</span><span class='op'>^</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='op'>-</span><span class='fl'>2</span>, <span class='fl'>3</span>, by <span class='op'>=</span> <span class='fl'>0.02</span><span class='op'>)</span>, cv <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.01</span>, <span class='fl'>0.1</span>, <span class='fl'>1</span>, <span class='fl'>3</span>, <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    <span class='co'># Standard gamma params</span></span>
<span>    scale <span class='op'>=</span> <span class='va'>cv</span><span class='op'>^</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>mean</span>,</span>
<span>    shape <span class='op'>=</span> <span class='fl'>1</span> <span class='op'>/</span> <span class='va'>cv</span><span class='op'>^</span><span class='fl'>2</span>,</span>
<span>    <span class='co'># neg binom params</span></span>
<span>    mu <span class='op'>=</span> <span class='va'>mean</span>,</span>
<span>    size <span class='op'>=</span> <span class='va'>shape</span></span>
<span>  <span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span></span>
<span>    prob_0 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>dnbinom</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='fl'>0</span>, mu <span class='op'>=</span> <span class='va'>mean</span>, size <span class='op'>=</span> <span class='va'>shape</span><span class='op'>)</span>,</span>
<span>    prob_ge_1 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>pnbinom</a></span><span class='op'>(</span>q <span class='op'>=</span> <span class='fl'>0</span>, mu <span class='op'>=</span> <span class='va'>mean</span>, size <span class='op'>=</span> <span class='va'>shape</span>, lower.tail <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,</span>
<span>    prob_ge_10 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>pnbinom</a></span><span class='op'>(</span>q <span class='op'>=</span> <span class='fl'>9</span>, mu <span class='op'>=</span> <span class='va'>mean</span>, size <span class='op'>=</span> <span class='va'>shape</span>, lower.tail <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>,</span>
<span>  <span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>clrs</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span>name<span class='op'>=</span><span class='st'>"Greens"</span>, n<span class='op'>=</span><span class='fl'>5</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span></span>
<span><span class='va'>nb_prob</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>cv</span> <span class='op'>&lt;</span> <span class='fl'>10</span>, <span class='va'>mean</span> <span class='op'>&lt;=</span> <span class='fl'>1e2</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> </span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>cv</span>, <span class='va'>as.factor</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>mean</span>, <span class='va'>prob_0</span>, color <span class='op'>=</span> <span class='va'>cv</span>, linetype <span class='op'>=</span> <span class='va'>cv</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='co'># scale_y_log10(limits = c(1e-10, 1)) +</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span></span>
<span>    <span class='co'># axis.text.y = element_blank(),</span></span>
<span>    <span class='co'># axis.ticks.y = element_blank()</span></span>
<span>  <span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='co'># scale_color_brewer(type = 'seq', palette = 5,</span></span>
<span>  <span class='co'>#                    values = ~scales::rescale(.x, to = c(0, 0.5))) +</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>'Pr(X = 0)'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>0.8</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-21-1.svg" width="1950" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>clrs</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span>name<span class='op'>=</span><span class='st'>"Greens"</span>, n<span class='op'>=</span><span class='fl'>5</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span></span>
<span><span class='va'>nb_prob</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>cv</span> <span class='op'>&lt;</span> <span class='fl'>10</span>, <span class='va'>mean</span> <span class='op'>&lt;=</span> <span class='fl'>1e2</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> </span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>cv</span>, <span class='va'>as.factor</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>mean</span>, <span class='va'>prob_ge_1</span>, color <span class='op'>=</span> <span class='va'>cv</span>, linetype <span class='op'>=</span> <span class='va'>cv</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span></span>
<span>    <span class='co'># axis.text.y = element_blank(),</span></span>
<span>    <span class='co'># axis.ticks.y = element_blank()</span></span>
<span>  <span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='co'># scale_color_brewer(type = 'seq', palette = 5,</span></span>
<span>  <span class='co'>#                    values = ~scales::rescale(.x, to = c(0, 0.5))) +</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>'Pr(X &gt;= 1)'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fl'>0.5</span>, color <span class='op'>=</span> <span class='st'>'grey'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>0.8</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-22-1.svg" width="1950" /></p>
</div>
<p>See that, for probability of 0 reads, a CV of 0.1 or less is indistinguishable from Poisson, but a CV on the order of 1 or more has a significant effect.
The point where detection becomes more likely than not (grey line) is similar for CV &lt;= 1, but becomes much larger for CV = 3 (about 100-fold larger).
These plots also show that there is a monotonic relationship of CV with probability of detection via a single read, where a larger CV lowers the probabability for every mean.</p>
<p>Next, consider the probability of detection in 10 or more reads,</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>clrs</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span>name<span class='op'>=</span><span class='st'>"Greens"</span>, n<span class='op'>=</span><span class='fl'>5</span><span class='op'>)</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span></span>
<span><span class='va'>nb_prob</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>cv</span> <span class='op'>&lt;</span> <span class='fl'>10</span>, <span class='va'>mean</span> <span class='op'>&gt;=</span> <span class='fl'>1e-1</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span> </span>
<span>  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>cv</span>, <span class='va'>as.factor</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>mean</span>, <span class='va'>prob_ge_10</span>, color <span class='op'>=</span> <span class='va'>cv</span>, linetype <span class='op'>=</span> <span class='va'>cv</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1e-4</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a></span><span class='op'>(</span></span>
<span>    <span class='co'># axis.text.y = element_blank(),</span></span>
<span>    <span class='co'># axis.ticks.y = element_blank()</span></span>
<span>  <span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>values <span class='op'>=</span> <span class='va'>clrs</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='co'># scale_color_brewer(type = 'seq', palette = 5,</span></span>
<span>  <span class='co'>#                    values = ~scales::rescale(.x, to = c(0, 0.5))) +</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>'Pr(X &gt;= 10)'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fl'>0.5</span>, color <span class='op'>=</span> <span class='st'>'grey'</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_path.html'>geom_line</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>0.8</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<p><img src="basic-detection-theory_files/figure-html5/unnamed-chunk-23-1.svg" width="1950" /></p>
</div>
<p>Here, the relationship of detection probability vs CV is more complicated, and depends on the mean.
At low means, where detection is extremely unlikely, then the larger CV helps by making these unlikely tail events more probable.
Fairly unprobably events, on the order of say 1e-2 or 1e-3, may still be observed when we are testing hundreds or thousands of samples.
Regarding when detection becomes more likely than not (grey line), we see something similar to single-read detection.
There is only a small increase for CV of 1 relative to Poisson, and a roughly 100-fold increase for CV of 3.</p>
<p>It should be easy to use the probability mass function of the neg binomial to examine how the mean and cv affect the probability of seeing 0 reads.</p>
<h4 id="reads-fully-covering-a-target-sub-sequence">Reads fully covering a target (sub-)sequence</h4>
<p>In some cases, we might be interested in how many reads that fully cover a particular subsequence, such as a particular 40-mer in the pathogen’s genome.</p>
<p>Supose we have</p>
<ul>
<li><span class="math inline">\(\mathcal M\)</span> total single-end reads, each of length <span class="math inline">\(l\)</span></li>
<li>A fraction <span class="math inline">\(P\)</span> are from the pathogen’s genome, or <span class="math inline">\(M = \mathcal M P\)</span> reads</li>
<li>The pathogen’s genome has length <span class="math inline">\(G\)</span> bp</li>
<li>We are interested in the number or fraction of reads that cover a particular k-mer (of lenth <span class="math inline">\(k\)</span>, with <span class="math inline">\(k \le l\)</span>).</li>
</ul>
<p>Reads that cover the k-mer are those whose start coincides with that of the k-mer or the <span class="math inline">\(l - k\)</span> bp before it.
Therefore, in the absence of within-genome bias in the read start, the fraction of pathogen reads that cover the focal k-mer is <span class="math inline">\((l - k + 1) / G\)</span>.
The expected number of reads covering a k-mer is (at least approximately)
<span class="math display">\[\begin{align}
  \mathcal M \cdot P \cdot \frac{l - k + 1}{G}
\end{align}\]</span>
Note, <span class="math inline">\(P\)</span> is itself expected to be proportional to the pathogen’s genome, so genome size is not expected to have an effect on the number of times we see a particular k-mer.</p>
<p>This equation tells us something about the type of sequencing we should prefer depending on the circumstances.
Suppose we are in a situation where reads that are at least 50bp long are sufficient to assign to the pathogen, and we just need to see the pathogen 10 times period to identify it.
In that case, we should prefer sequencing a sequencing option that maximizes the total number of reads, provided the reads are 50bp or longer.
(This observation explains why you sometimes read/here that it is better to go with shorter-read options for certain ‘counting’ applications, such as gene expression studies)</p>
<p>But if we want to see a particular k-mer, then we want to maximize <span class="math inline">\(\mathcal M (l - k + 1)\)</span>.
For <span class="math inline">\(l \gg k\)</span>, this looks like <span class="math inline">\(\mathcal M l\)</span>, or the total number of sequenced basepairs; at the values of <span class="math inline">\(l\)</span> we are often considering with Illumina sequencing, there is also an additional premium beyond this on having longer reads, due to the need to completely cover the k-mer.</p>
<p>For Illumina sequencing, the sequencing options that maximize total bp also tend to maximize total read length.
There are options that give more reads, but shorter reads with total bp.
These options may make sense for certain applications, but I suspect we’ll often want longer reads as long as this maximizes total bp, as long as we can generate sequencing libraries with sufficiently long inserts.</p>
<p>I believe this same basic tradeoff/choice behavior also occurs with Ultima Genomics’ sequencing platform.
With Ultima’s platform, you decide how many cycles (bp) to sequence.
Running for less cycles gets you shorter reads, but is cheaper per read, though not per bp.</p>
<p>I have less of a handle on how this works with ONT, though I wouldn’t be surprised if a similar trade also exists there.</p>
<p>TODO: find my notes from Metasub on this topic</p>
<h3 id="targeted-detection-using-qpcr-or-dpcr">Targeted detection using qPCR or dPCR</h3>
<p>In future, I’ll add notes on qPCR and dPCR, as standard methods for targeted detection of pathogens.</p>
<h3 id="amplicon-sequencing">Amplicon sequencing</h3>
<p>In future, I’ll add some notes on amplicon sequencing, which is the primary method used for sequencing SARS-CoV-2 in wastewater (to identify and monitor variants).</p>
<ul>
<li>Amplicon sequencing involves using PCR to amplify one or more regions of the target organism’s genome, and then sequencing the resulting amplicons.</li>
<li>For SARS-CoV-2, it is common to use a tiling amplicon approach, where the genome is divided into overlapping amplicons of a fixed length, and each amplicon is amplified and sequenced simultaneously.</li>
<li>We are interested in questions such as at what point in its spread do we detect a new mutation or variant. The key difference from MGS is that what matters for amplicon sequencing is the relative abundance of the variant relative to the amplified genomes, rather than all genomes.</li>
<li>The same more detailed MGS model, described in my previous work, can be used to derive appropriate simplified models for MGS and amplicon sequencing, by appropriate choice of the efficiency parameters</li>
<li>For the simplified amplicon model, we can restrict the ‘background’ to just the target species (assuming negligible off-target amplification). This amounts to setting <span class="math inline">\(B=0\)</span> for other species. We need to consider bias among variants within the target species, which can arise for example due to mutations in the primer-binding regions.</li>
</ul>
<h1 id="notes">Notes</h1>
<h2 id="transportation-time">Transportation time</h2>
<h2 id="effective-number-of-shedders-contributing-to-a-wwtp-sample">Effective number of shedders contributing to a WWTP sample</h2>
<h2 id="non-human-contributions-to-wwtp-samples">Non-human contributions to WWTP samples</h2>
<p>Above, I assume that the WWTP samples are made up entirely from human microbiome shedding.
But there may be other contributions to the sample, including</p>
<ul>
<li>contributions of microbial and non-microbial NA from other sources, such as pets, livestock, or wildlife, flushed or carried in by runoff or stormwater</li>
<li>growth in the sewer system (including in the primary clarifier, for settled solids)</li>
</ul>
<p>TODO: consider whether it is plausible that these contribute a signficant relative abundance of MGS reads.
For growth in the sewers, review a recent college campus study that suggested this was a significant facter in campus samples at the start of the school year, when the system had been in low use.
Can also look for literature on microbial dynamics in clarifiers, and talk to the scientists who study this stuff, for the case of settled solids.</p>
<h2 id="noisevariability-in-the-model">Noise/variability in the model</h2>
<p>There is noise or randomness injected from many sources including transmission, shedding, sample processing, and sequencing.
My working hypothesis is that variability for basic detection in the WWTP setting is negligible because we’ll be detecting the epidemic at a sufficiently late point in time in a sufficiently large catchment that the random effects will largely average out in predictions of the point when detection occurs.
It will be important to assess the relevant parameters and see if/when we are likely in a regime where this hypothesis holds.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-diekmann2012math" class="csl-entry" role="listitem">
Diekmann, Odo, Hans Heesterbeek, and Tom Britton. 2012. <em>Mathematical <span>Tools</span> for <span>Understanding Infectious Disease Dynamics</span></em>. First. <span>Princeton University Press</span>. <a href="https://doi.org/10.23943/princeton/9780691155395.001.0001">https://doi.org/10.23943/princeton/9780691155395.001.0001</a>.
</div>
<div id="ref-heng2020theaa" class="csl-entry" role="listitem">
Heng, Kevin, and Christian L. Althaus. 2020. <span>“The Approximately Universal Shapes of Epidemic Curves in the <span>Susceptible-Exposed-Infectious-Recovered</span> (<span>SEIR</span>) Model.”</span> <em>Scientific Reports</em> 10 (1): 19365. <a href="https://doi.org/10.1038/s41598-020-76563-8">https://doi.org/10.1038/s41598-020-76563-8</a>.
</div>
<div id="ref-kapo2017esti" class="csl-entry" role="listitem">
Kapo, Katherine E., Michael Paschka, Raghu Vamshi, Megan Sebasky, and Kathleen McDonough. 2017. <span>“Estimation of <span>U</span>.<span>S</span>. Sewer Residence Time Distributions for National-Scale Risk Assessment of down-the-Drain Chemicals.”</span> <em>Science of The Total Environment</em> 603–604 (December): 445–52. <a href="https://doi.org/10.1016/j.scitotenv.2017.06.075">https://doi.org/10.1016/j.scitotenv.2017.06.075</a>.
</div>
<div id="ref-kim2022sars" class="csl-entry" role="listitem">
Kim, Sooyeol, Lauren C. Kennedy, Marlene K. Wolfe, Craig S. Criddle, Dorothea H. Duong, Aaron Topol, Bradley J. White, et al. 2022. <span>“<span>SARS-CoV-2 RNA</span> Is Enriched by Orders of Magnitude in Primary Settled Solids Relative to Liquid Wastewater at Publicly Owned Treatment Works.”</span> <em>Environmental Science: Water Research &amp; Technology</em> 8 (4): 757–70. <a href="https://doi.org/10.1039/D1EW00826A">https://doi.org/10.1039/D1EW00826A</a>.
</div>
<div id="ref-ma2020esti" class="csl-entry" role="listitem">
Ma, Junling. 2020. <span>“Estimating Epidemic Exponential Growth Rate and Basic Reproduction Number.”</span> <em>Infectious Disease Modelling</em> 5 (January): 129–41. <a href="https://doi.org/10.1016/j.idm.2019.12.009">https://doi.org/10.1016/j.idm.2019.12.009</a>.
</div>
<div id="ref-morantovar2022stoc" class="csl-entry" role="listitem">
Morán-Tovar, Roberto, Henning Gruell, Florian Klein, and Michael Lässig. 2022. <span>“Stochasticity of Infectious Outbreaks and Consequences for Optimal Interventions.”</span> <em>Journal of Physics A: Mathematical and Theoretical</em> 55 (38): 384008. <a href="https://doi.org/10.1088/1751-8121/ac88a6">https://doi.org/10.1088/1751-8121/ac88a6</a>.
</div>
<div id="ref-rothman2021rnav" class="csl-entry" role="listitem">
Rothman, Jason A., Theresa B. Loveless, Joseph Kapcia, Eric D. Adams, Joshua A. Steele, Amity G. Zimmer-Faust, Kylie Langlois, et al. 2021. <span>“<span>RNA Viromics</span> of <span>Southern California Wastewater</span> and <span>Detection</span> of <span>SARS-CoV-2 Single-Nucleotide Variants</span>.”</span> <em>Applied and Environmental Microbiology</em> 87 (23): e01448–21. <a href="https://doi.org/10.1128/AEM.01448-21">https://doi.org/10.1128/AEM.01448-21</a>.
</div>
<div id="ref-soller2022mode" class="csl-entry" role="listitem">
Soller, Jeffrey, Wiley Jennings, Mary Schoen, Alexandria Boehm, Krista Wigginton, Raul Gonzalez, Katherine E. Graham, Graham McBride, Amy Kirby, and Mia Mattioli. 2022. <span>“Modeling Infection from <span>SARS-CoV-2</span> Wastewater Concentrations: Promise, Limitations, and Future Directions.”</span> <em>Journal of Water and Health</em> 20 (8): 1197–1211. <a href="https://doi.org/10.2166/wh.2022.094">https://doi.org/10.2166/wh.2022.094</a>.
</div>
</div>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr />
<ol>
<li id="fn1"><p><a href="https://www.randomservices.org/random/brown/Drift.html" class="uri">https://www.randomservices.org/random/brown/Drift.html</a><a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
  <h3 id="references">References</h3>
  <div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
